Deity

juxjux9930 opened the pull request:
🆑 Jux
rscadd: Re-adds Deity.
/🆑

I'm not like, doing better, but it was almost done last time.

The big bug was supposedly something about the deity vision not working, but I'm having difficulty consistently reproducing it provided I juggle my consciousness correctly. What I _can_ reproduce is that the construction spells seem to always fizzle out due to a lack of targets, but this may be because I can't be the God that it wants to pull power from & the Cultist building things at the same time. Idk.

 Need a LOT of help testing things and some of this testing may need to be done on live.

juxtaposed9930 commented:
Thank you Sierra and Mucker for helping me pass checks

SierraKomodo commented on code/_onclick/hud/deity.dm:
```diff
@@ -0,0 +1,57 @@
+
+/datum/hud/deity/FinalizeInstantiation(ui_style = 'icons/mob/screen1_Midnight.dmi')
+	src.adding = list()
+	src.other = list()
```
`src.` is redundant here - It's only needed if a local var or parameter exists in the proc with the same name.

```suggestion
	adding = list()
	other = list()
```

SierraKomodo commented on code/_onclick/hud/deity.dm:
```diff
@@ -0,0 +1,57 @@
+
+	var/obj/screen/intent/deity/D = new()
+
+	src.adding += D
```
```suggestion
	adding += D
```

SierraKomodo commented on code/_onclick/hud/deity.dm:
```diff
@@ -0,0 +1,57 @@
+	action_intent = D
+
+	mymob.client.screen = list()
+	mymob.client.screen += src.adding
```
```suggestion
	mymob.client.screen += adding
```

SierraKomodo commented on code/_onclick/hud/deity.dm:
```diff
@@ -0,0 +1,57 @@
+	..()
+	overlays += image('icons/mob/screen_phenomena.dmi', icon_state = "hud", pixel_x = -138, pixel_y = -1)
+
+/obj/screen/intent/deity/proc/sync_to_mob(var/mob)
```
`/var` is redundant in parameters

```suggestion
/obj/screen/intent/deity/proc/sync_to_mob(mob)
```

SierraKomodo commented on code/_onclick/hud/deity.dm:
```diff
@@ -0,0 +1,57 @@
+	overlays += image('icons/mob/screen_phenomena.dmi', icon_state = "hud", pixel_x = -138, pixel_y = -1)
+
+/obj/screen/intent/deity/proc/sync_to_mob(var/mob)
+	var/mob/living/deity/D = mob
```
If we know that `mob` will always be a deity, just set the type in the parameter itself. If there's the possibility it's not a deity, this needs a type check for sanity reasons.

juxjux9930 replied:
considering most of this is a revert to old code, just for clarification- setting the type in the parameter would look like 
`/obj/screen/intent/deity/proc/sync_to_mob(/mob/living/deity/D)`
or something else?

SierraKomodo replied:
`mob/living/deity/D` - No leading slash but otherwise correct

SierraKomodo commented on code/_onclick/hud/deity.dm:
```diff
@@ -0,0 +1,57 @@
+		else
+			S.maptext = null
+
+/obj/screen/intent/deity/Click(var/location, var/control, var/params)
```
```suggestion
/obj/screen/intent/deity/Click(location, control, params)
```

SierraKomodo commented on code/datums/extensions/deity_be_near.dm:
```diff
@@ -0,0 +1,67 @@
+	var/expected_helmet
+	flags = EXTENSION_FLAG_IMMEDIATE
+
+/datum/extension/deity_be_near/New(var/datum/holder, var/mob/living/deity/connect)
```
```suggestion
/datum/extension/deity_be_near/New(datum/holder, mob/living/deity/connect)
```

SierraKomodo commented on code/datums/extensions/deity_be_near.dm:
```diff
@@ -0,0 +1,67 @@
+		deal_damage(I.loc, round(threshold_base/min_dist))
+
+
+/datum/extension/deity_be_near/proc/deal_damage(var/mob/living/victim, var/mult)
```
```suggestion
/datum/extension/deity_be_near/proc/deal_damage(mob/living/victim, mult)
```

SierraKomodo commented on code/datums/extensions/deity_be_near.dm:
```diff
@@ -0,0 +1,67 @@
+
+/datum/extension/deity_be_near/proc/dead_deity()
+	var/obj/item/I = holder
+	I.visible_message("<span class='warning'>\The [holder]'s power fades!</span>")
```
```suggestion
	I.visible_message(SPAN_WARNING("\The [holder]'s power fades!"))
```

SierraKomodo commented on code/datums/extensions/deity_be_near.dm:
```diff
@@ -0,0 +1,67 @@
+		return FALSE
+	return TRUE
+
+/datum/extension/deity_be_near/champion/deal_damage(var/mob/living/victim,var/mult)
```
```suggestion
/datum/extension/deity_be_near/champion/deal_damage(mob/living/victim, mult)
```

SierraKomodo commented on code/datums/extensions/deity_be_near.dm:
```diff
@@ -0,0 +1,67 @@
+/datum/extension/deity_be_near/champion/deal_damage(var/mob/living/victim,var/mult)
+	victim.adjustOxyLoss(3 * mult)
+
+/datum/extension/deity_be_near/oracle/deal_damage(var/mob/living/victim, var/mult)
```
```suggestion
/datum/extension/deity_be_near/oracle/deal_damage(mob/living/victim, mult)
```

SierraKomodo commented on code/datums/extensions/deity_be_near.dm:
```diff
@@ -0,0 +1,67 @@
+/datum/extension/deity_be_near/oracle/deal_damage(var/mob/living/victim, var/mult)
+	victim.adjustFireLoss(mult)
+
+/datum/extension/deity_be_near/traitor/deal_damage(var/mob/living/victim, var/mult)
```
```suggestion
/datum/extension/deity_be_near/traitor/deal_damage(mob/living/victim, mult)
```

SierraKomodo commented on code/game/antagonist/outsider/deity.dm:
```diff
@@ -0,0 +1,18 @@
+	role_text = "Deity"
+	role_text_plural = "Deities"
+	mob_path = /mob/living/deity
+	welcome_text = "This is not your world. This is not your reality. But here you exist. Use your powers, feed off the faith of others.<br>You have to click on yourself to choose your form.<br>Everything you say will be heard by your cultists!<br>To get points your cultists need to build!<br>Build Shrine and Construction are the best starting boons!"
```
```suggestion
	welcome_text = "This is not your world. This is not your reality. But here you exist. Use your powers, feed off the faith of others.<br>You have to click on yourself to choose your form.<br>Everything you say will be heard by your cultists!<br>To get points, your cultists need to build!<br>Build Shrine and Construction are the best starting boons!"
```

SierraKomodo commented on code/game/antagonist/station/cult_god.dm:
```diff
@@ -0,0 +1,112 @@
+	antaghud_indicator = "hudcultist"
+	skill_setter = /datum/antag_skill_setter/station
+
+/datum/antagonist/godcultist/add_antagonist_mind(var/datum/mind/player, var/ignore_role, var/nonstandard_role_type, var/nonstandard_role_msg, var/mob/living/deity/specific_god)
```
```suggestion
/datum/antagonist/godcultist/add_antagonist_mind(datum/mind/player, ignore_role, nonstandard_role_type, nonstandard_role_msg, mob/living/deity/specific_god)
```

SierraKomodo commented on code/game/antagonist/station/cult_god.dm:
```diff
@@ -0,0 +1,112 @@
+
+/datum/antagonist/godcultist/add_antagonist_mind(var/datum/mind/player, var/ignore_role, var/nonstandard_role_type, var/nonstandard_role_msg, var/mob/living/deity/specific_god)
+	if(!..())
+		return 0
```
```suggestion
		return FALSE
```

SierraKomodo commented on code/game/antagonist/station/cult_god.dm:
```diff
@@ -0,0 +1,112 @@
+	if(specific_god)
+		add_cultist(player, specific_god)
+
+	return 1
```
```suggestion
	return TRUE
```

SierraKomodo commented on code/game/antagonist/station/cult_god.dm:
```diff
@@ -0,0 +1,112 @@
+	return 1
+
+/datum/antagonist/godcultist/post_spawn()
+	if(!GLOB.deity || !GLOB.deity.current_antagonists.len)
```
```suggestion
	if(!GLOB.deity?.current_antagonists.len)
```

SierraKomodo commented on code/game/antagonist/station/cult_god.dm:
```diff
@@ -0,0 +1,112 @@
+		deity_count++
+
+
+/datum/antagonist/godcultist/remove_antagonist(var/datum/mind/player, var/show_message, var/implanted)
```
```suggestion
/datum/antagonist/godcultist/remove_antagonist(datum/mind/player, show_message, implanted)
```

SierraKomodo commented on code/game/antagonist/station/cult_god.dm:
```diff
@@ -0,0 +1,112 @@
+/datum/antagonist/godcultist/remove_antagonist(var/datum/mind/player, var/show_message, var/implanted)
+	var/mob/living/deity/god = get_deity(player)
+	if(!..())
+		return 0
```
```suggestion
		return FALSE
```

SierraKomodo commented on code/game/antagonist/station/cult_god.dm:
```diff
@@ -0,0 +1,112 @@
+	if(!..())
+		return 0
+	remove_cultist(player, god)
+	return 1
```
```suggestion
	return TRUE
```

SierraKomodo commented on code/game/antagonist/station/cult_god.dm:
```diff
@@ -0,0 +1,112 @@
+	remove_cultist(player, god)
+	return 1
+
+/datum/antagonist/godcultist/get_extra_panel_options(var/datum/mind/player)
```
```suggestion
/datum/antagonist/godcultist/get_extra_panel_options(datum/mind/player)
```

SierraKomodo commented on code/game/antagonist/station/cult_god.dm:
```diff
@@ -0,0 +1,112 @@
+
+/datum/antagonist/godcultist/Topic(href, href_list)
+	if(..())
+		return 1
```
```suggestion
		return TRUE
```

SierraKomodo commented on code/game/antagonist/station/cult_god.dm:
```diff
@@ -0,0 +1,112 @@
+				log_and_message_admins("has set [key_name(player.current)] to be a minion of [key_name(D)]")
+		else
+			to_chat(usr, SPAN_WARNING("There are no deities to be linked to."))
+		return 1
```
```suggestion
		return TRUE
```

SierraKomodo commented on code/game/antagonist/station/cult_god.dm:
```diff
@@ -0,0 +1,112 @@
+			to_chat(usr, SPAN_WARNING("There are no deities to be linked to."))
+		return 1
+
+/datum/antagonist/godcultist/proc/add_cultist(var/datum/mind/player, var/mob/living/deity/deity)
```
```suggestion
/datum/antagonist/godcultist/proc/add_cultist(datum/mind/player, mob/living/deity/deity)
```

SierraKomodo commented on code/game/antagonist/station/cult_god.dm:
```diff
@@ -0,0 +1,112 @@
+	deity.add_follower(player.current)
+	player.current.add_language(LANGUAGE_CULT)
+
+/datum/antagonist/godcultist/proc/remove_cultist(var/datum/mind/player, var/mob/living/deity/god)
```
```suggestion
/datum/antagonist/godcultist/proc/remove_cultist(datum/mind/player, mob/living/deity/god)
```

SierraKomodo commented on code/game/antagonist/station/cult_god.dm:
```diff
@@ -0,0 +1,112 @@
+	god.remove_follower(player.current)
+	player.current.remove_language(LANGUAGE_CULT)
+
+/datum/antagonist/godcultist/proc/get_deity(var/datum/mind/player)
```
```suggestion
/datum/antagonist/godcultist/proc/get_deity(datum/mind/player)
```

SierraKomodo commented on code/game/antagonist/station/cult_god.dm:
```diff
@@ -0,0 +1,112 @@
+
+/datum/antagonist/godcultist/proc/get_deity(var/datum/mind/player)
+	for(var/m in GLOB.deity.current_antagonists)
+		var/datum/mind/mind = m
```
```suggestion
	for(var/datum/mind/mind as anything in GLOB.deity.current_antagonists)
```

SierraKomodo commented on code/game/antagonist/station/cult_god.dm:
```diff
@@ -0,0 +1,112 @@
+	for(var/m in GLOB.deity.current_antagonists)
+		var/datum/mind/mind = m
+		var/mob/living/deity/god = mind.current
+		if(god && god.is_follower(player.current,1))
```
```suggestion
		if(god?.is_follower(player.current,1))
```

SierraKomodo commented on code/game/antagonist/station/cult_god.dm:
```diff
@@ -0,0 +1,112 @@
+		if(god && god.is_follower(player.current,1))
+			return god
+
+/mob/living/proc/dpray(var/msg as text)
```
```suggestion
/mob/living/proc/dpray(msg as text)
```

SierraKomodo commented on code/game/antagonist/station/cult_god.dm:
```diff
@@ -0,0 +1,112 @@
+/mob/living/proc/dpray(var/msg as text)
+	set category = "Abilities"
+
+	if(!src.mind || !GLOB.godcult || !GLOB.godcult.is_antagonist(mind))
```
```suggestion
	if(!mind || !GLOB.godcult?.is_antagonist(mind))
```

SierraKomodo commented on code/game/antagonist/station/cult_god.dm:
```diff
@@ -0,0 +1,112 @@
+		return
+
+	//Make em wait a few seconds.
+	src.visible_message("\The [src] bows their head down, muttering something.", SPAN_NOTICE("You send the message \"[msg]\" to your master."))
```
```suggestion
	visible_message(
		"\The [src] bows their head down, muttering something.",
		SPAN_NOTICE("You send the message \"[msg]\" to your master.")
	)
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/narsie_items.dm:
```diff
@@ -0,0 +1,93 @@
+	desc = "This knife is dull but well used."
+	default_material = MATERIAL_CULT
+
+/obj/item/material/knife/ritual/sacrifice/resolve_attackby(var/atom/a, var/mob/user, var/click_params)
```
```suggestion
/obj/item/material/knife/ritual/sacrifice/resolve_attackby(atom/a, mob/user, click_params)
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/narsie_items.dm:
```diff
@@ -0,0 +1,93 @@
+			return
+
+		user.visible_message(SPAN_WARNING("\The [user] hovers \the [src] over \the [a], whispering an incantation."))
+		if(!do_after(user,200, L))
```
```suggestion
		if(!do_after(user, 20 SECONDS, L))
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/narsie_items.dm:
```diff
@@ -0,0 +1,93 @@
+			to_chat(user, SPAN_WARNING("\The [a] is already dead! There is nothing to take!"))
+			return
+
+		user.visible_message(SPAN_WARNING("\The [user] hovers \the [src] over \the [a], whispering an incantation."))
```
Or however you wish to fluff the message sent to the user.

```suggestion
		user.visible_message(
			SPAN_WARNING("\The [user] hovers \the [src] over \the [a], whispering an incantation."),
			SPAN_WARNING("You hover \the [src] over \the [a], whispering an incantation.")
		)
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/narsie_items.dm:
```diff
@@ -0,0 +1,93 @@
+		user.visible_message(SPAN_WARNING("\The [user] hovers \the [src] over \the [a], whispering an incantation."))
+		if(!do_after(user,200, L))
+			return
+		user.visible_message(SPAN_DANGER("\The [user] plunges the knife down into \the [a]!"))
```
```suggestion
		user.visible_message(
			SPAN_DANGER("\The [user] plunges \the [src] down into \the [a]!"),
			SPAN_DANGER("You plunge \the [src] down into \the [a]!")
		)
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/narsie_items.dm:
```diff
@@ -0,0 +1,93 @@
+		user.visible_message(SPAN_DANGER("\The [user] plunges the knife down into \the [a]!"))
+		L.adjustBruteLoss(20)
+		if(altar.linked_god)
+			altar.linked_god.adjust_power_min(2 * multiplier,0,"from a delicious sacrifice!")
```
```suggestion
			altar.linked_god.adjust_power_min(2 * multiplier, FALSE, "from a delicious sacrifice!")
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/narsie_items.dm:
```diff
@@ -0,0 +1,93 @@
+	if(stored_power)
+		to_chat(user, SPAN_NOTICE("It exudes a death-like smell."))
+
+/obj/item/material/twohanded/fireaxe/cult/resolve_attackby(var/atom/a, var/mob/user, var/click_params)
```
```suggestion
/obj/item/material/twohanded/fireaxe/cult/resolve_attackby(atom/a, mob/user, click_params)
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/narsie_items.dm:
```diff
@@ -0,0 +1,93 @@
+
+/obj/item/material/twohanded/fireaxe/cult/proc/gain_power()
+	stored_power += 50
+	src.visible_message(SPAN_OCCULT("\The [src] screeches as the smell of death fills the air!"))
```
```suggestion
	visible_message(SPAN_OCCULT("\The [src] screeches as the smell of death fills the air!"))
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/narsie_structures.dm:
```diff
@@ -0,0 +1,85 @@
+
+/obj/structure/deity/blood_forge
+	name = "unholy forge"
+	desc = "This forge gives off no heat, no light, its flames look almost unnatural."
```
```suggestion
	desc = "This forge gives off no heat, no light; its flames look almost unnatural."
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/narsie_structures.dm:
```diff
@@ -0,0 +1,85 @@
+
+	power_adjustment = 2
+
+/obj/structure/deity/blood_forge/attack_hand(var/mob/user)
```
```suggestion
/obj/structure/deity/blood_forge/attack_hand(mob/user)
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/narsie_structures.dm:
```diff
@@ -0,0 +1,85 @@
+	power_adjustment = 2
+
+/obj/structure/deity/blood_forge/attack_hand(var/mob/user)
+	if(!linked_god || !linked_god.is_follower(user, silent = 1) || !ishuman(user))
```
```suggestion
	if(!linked_god?.is_follower(user, silent = TRUE) || !ishuman(user))
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/narsie_structures.dm:
```diff
@@ -0,0 +1,85 @@
+
+	var/dat = "<center><b>Recipies</b></center><br><br><i>Item - [text_modifications["Cost"]] Cost</i><br>"
+	for(var/type in recipes)
+		var/atom/a = type
```
```suggestion
	for(var/atom/a as anything in recipes)
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/narsie_structures.dm:
```diff
@@ -0,0 +1,85 @@
+		dat += "<A href='?src=\ref[src];make_recipe=\ref[type];'>[initial(a.name)]</a> - [cost]<br><i>[initial(a.desc)]</i><br><br>"
+	show_browser(user, dat, "window=forge")
+
+/obj/structure/deity/blood_forge/CanUseTopic(var/user)
```
```suggestion
/obj/structure/deity/blood_forge/CanUseTopic(user)
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/narsie_structures.dm:
```diff
@@ -0,0 +1,85 @@
+	show_browser(user, dat, "window=forge")
+
+/obj/structure/deity/blood_forge/CanUseTopic(var/user)
+	if(!linked_god || !linked_god.is_follower(user, silent = 1) || !ishuman(user))
```
```suggestion
	if(!linked_god?.is_follower(user, silent = TRUE) || !ishuman(user))
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/narsie_structures.dm:
```diff
@@ -0,0 +1,85 @@
+		return STATUS_CLOSE
+	return ..()
+
+/obj/structure/deity/blood_forge/OnTopic(var/user, var/list/href_list)
```
```suggestion
/obj/structure/deity/blood_forge/OnTopic(user, list/href_list)
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/narsie_structures.dm:
```diff
@@ -0,0 +1,85 @@
+			craft_item(type, cost, user)
+		return TOPIC_REFRESH
+
+/obj/structure/deity/blood_forge/proc/craft_item(var/path, var/blood_cost, var/mob/user)
```
```suggestion
/obj/structure/deity/blood_forge/proc/craft_item(path, blood_cost, mob/user)
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/narsie_structures.dm:
```diff
@@ -0,0 +1,85 @@
+		if(linked_god)
+			linked_god.take_charge(user, 10)
+	var/obj/item/I = new path(get_turf(src))
+	user.visible_message("\The [user] pull out \the [I] from the [text_modifications["Out"]].", "You pull out the completed [I] from the [text_modifications["Out"]].")
```
```suggestion
	user.visible_message(
		SPAN_NOTICE("\The [user] pulls \an [I] from the [text_modifications["Out"]]."),
		SPAN_NOTICE("You pull the completed [I] from the [text_modifications["Out"]].")
	)
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/narsie_structures.dm:
```diff
@@ -0,0 +1,85 @@
+	var/obj/item/I = new path(get_turf(src))
+	user.visible_message("\The [user] pull out \the [I] from the [text_modifications["Out"]].", "You pull out the completed [I] from the [text_modifications["Out"]].")
+
+/obj/structure/deity/blood_forge/proc/take_charge(var/mob/living/user, var/charge)
```
```suggestion
/obj/structure/deity/blood_forge/proc/take_charge(mob/living/user, charge)
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/narsie_structures.dm:
```diff
@@ -0,0 +1,85 @@
+	for(var/count = 0, count < blood_cost/10, count++)
+		if(!do_after(user, 5 SECONDS, src, DO_DEFAULT | DO_TARGET_UNIQUE_ACT))
+			return
+		user.visible_message("\The [user] swirls their hands in \the [src].", text_modifications["Shape"])
```
```suggestion
		user.visible_message(
			SPAN_NOTICE("\The [user] swirls their hands in \the [src]."),
			SPAN_NOTIC(text_modifications["Shape"])
		)
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/narsie_structures.dm:
```diff
@@ -0,0 +1,85 @@
+	build_cost = 700
+
+/obj/structure/deity/blood_stone/attack_hand(var/mob/user)
+	if(!linked_god || !linked_god.is_follower(user, silent = 1) || !ishuman(user))
```
```suggestion
	if(!linked_god?.is_follower(user, silent = TRUE) || !ishuman(user))
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/narsie_structures.dm:
```diff
@@ -0,0 +1,85 @@
+		return
+
+	var/mob/living/carbon/human/H = user
+	user.visible_message(SPAN_WARNING("\The [user] calmly slices their finger on \the [src], smeering it over the black stone."),SPAN_WARNING("You slowly slide your finger down one of \the [src]'s sharp edges, smeering it over its smooth surface."))
```
```suggestion
	user.visible_message(
		SPAN_WARNING("\The [user] calmly slices their finger on \the [src], smearing it over the black stone."),
		SPAN_WARNING("You slowly slide your finger down one of \the [src]'s sharp edges, smearing it over its smooth surface.")
	)
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/narsie_structures.dm:
```diff
@@ -0,0 +1,85 @@
+
+	var/mob/living/carbon/human/H = user
+	user.visible_message(SPAN_WARNING("\The [user] calmly slices their finger on \the [src], smeering it over the black stone."),SPAN_WARNING("You slowly slide your finger down one of \the [src]'s sharp edges, smeering it over its smooth surface."))
+	if(do_after(H, 50, src, DO_DEFAULT | DO_TARGET_UNIQUE_ACT))
```
```suggestion
	if(do_after(H, 5 SECONDS, src, DO_DEFAULT | DO_TARGET_UNIQUE_ACT))
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/narsie_structures.dm:
```diff
@@ -0,0 +1,85 @@
+	var/mob/living/carbon/human/H = user
+	user.visible_message(SPAN_WARNING("\The [user] calmly slices their finger on \the [src], smeering it over the black stone."),SPAN_WARNING("You slowly slide your finger down one of \the [src]'s sharp edges, smeering it over its smooth surface."))
+	if(do_after(H, 50, src, DO_DEFAULT | DO_TARGET_UNIQUE_ACT))
+		user.audible_message("\The [user] utters something under their breath.", SPAN_OCCULT("You mutter a dark prayer to your master as you feel the stone eat away at your lifeforce."))
```
```suggestion
		user.audible_message(
			"\The [user] utters something under their breath.",
			SPAN_OCCULT("You mutter a dark prayer to your master as you feel the stone eat away at your lifeforce.")
		)
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/starlight_items.dm:
```diff
@@ -0,0 +1,159 @@
+/obj/item/clothing/ring/aura_ring
+	var/obj/aura/granted_aura
+
+/obj/item/clothing/ring/aura_ring/equipped(var/mob/living/L, var/slot)
```
```suggestion
/obj/item/clothing/ring/aura_ring/equipped(mob/living/L, slot)
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/starlight_items.dm:
```diff
@@ -0,0 +1,159 @@
+	if(granted_aura && slot == slot_gloves)
+		L.add_aura(granted_aura)
+
+/obj/item/clothing/ring/aura_ring/dropped(var/mob/living/L)
```
```suggestion
/obj/item/clothing/ring/aura_ring/dropped(mob/living/L)
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/starlight_items.dm:
```diff
@@ -0,0 +1,159 @@
+	force_multiplier = 0.3
+	var/charge = 5
+
+/obj/item/material/knife/ritual/shadow/apply_hit_effect(var/mob/living/target, var/mob/living/user, var/hit_zone)
```
```suggestion
/obj/item/material/knife/ritual/shadow/apply_hit_effect(mob/living/target, mob/living/user, hit_zone)
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/starlight_items.dm:
```diff
@@ -0,0 +1,159 @@
+	else
+		user.adjustFireLoss(5)
+		if(prob(5))
+			to_chat(user, "<span class='warning'>\The [src] appears to be out of power!</span>")
```
```suggestion
			to_chat(user, SPAN_WARNING("\The [src] appears to be out of power!"))
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/starlight_items.dm:
```diff
@@ -0,0 +1,159 @@
+	desc = "Look closely into its crystal; there's a miniature sun. Or maybe that's just some fancy LEDs. Either way, it looks thoroughly mystical."
+	icon = 'icons/obj/wizard.dmi'
+	icon_state = "starstaff"
+	self_recharge = 0
```
```suggestion
	self_recharge = FALSE
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/starlight_items.dm:
```diff
@@ -0,0 +1,159 @@
+	var/last_near_structure = 0
+	var/mob/living/deity/linked
+
+/obj/item/material/sword/blazing/Initialize(var/maploading, var/material, var/deity)
```
```suggestion
/obj/item/material/sword/blazing/Initialize(maploading, material, deity)
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/starlight_items.dm:
```diff
@@ -0,0 +1,159 @@
+
+	if(linked.near_structure(src,1))
+		if(last_near_structure < world.time - 30 SECONDS)
+			to_chat(loc, "<span class='notice'>\The [src] surges with power anew!</span>")
```
```suggestion
			to_chat(loc, SPAN_NOTICE("\The [src] surges with power anew!"))
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/starlight_items.dm:
```diff
@@ -0,0 +1,159 @@
+	else
+		if(last_near_structure < world.time - 30 SECONDS) //If it has been at least 30 seconds.
+			if(prob(5))
+				to_chat(loc, "<span class='warning'>\The [src] begins to fade, its power dimming this far away from a shrine.</span>")
```
```suggestion
				to_chat(loc, SPAN_WARNING("\The [src] begins to fade, its power dimming this far away from a shrine."))
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/starlight_items.dm:
```diff
@@ -0,0 +1,159 @@
+			if(prob(5))
+				to_chat(loc, "<span class='warning'>\The [src] begins to fade, its power dimming this far away from a shrine.</span>")
+		else if(last_near_structure + 1800 < world.time)
+			visible_message("<span class='warning'>\The [src] disintegrates into a pile of ash!</span>")
```
```suggestion
			visible_message(SPAN_WARNING("\The [src] disintegrates into a pile of ash!"))
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/starlight_mobs.dm:
```diff
@@ -0,0 +1,33 @@
+	bone_material = null
+	bone_amount = 0
+
+/mob/living/starlight_soul/Initialize(var/maploading, var/mob/living/old_mob)
```
```suggestion
/mob/living/starlight_soul/Initialize(maploading, mob/living/old_mob)
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/starlight_mobs.dm:
```diff
@@ -0,0 +1,33 @@
+	if(old_mob)
+		name = old_mob.real_name
+
+/mob/living/starlight_soul/proc/set_deity(var/mob/living/deity/deity)
```
```suggestion
/mob/living/starlight_soul/proc/set_deity(mob/living/deity/deity)
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/starlight_mobs.dm:
```diff
@@ -0,0 +1,33 @@
+		eye.suffix = "Soul"
+		eyeobj = eye
+	eyeobj.visualnet = deity.eyeobj.visualnet
+	GLOB.godcult.add_antagonist_mind(src.mind,1,"lost soul of [deity]", "You have been captured by \the [deity]! You now can only see into your own reality through the same rips and tears it uses. Your only chance at another body will be one in your captor's image...",specific_god=deity)
```
```suggestion
	GLOB.godcult.add_antagonist_mind(mind, TRUE, "lost soul of [deity]", "You have been captured by \the [deity]! You now can only see into your own reality through the same rips and tears it uses. Your only chance at another body will be one in your captor's image...",specific_god=deity)
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/starlight_structures.dm:
```diff
@@ -0,0 +1,269 @@
+			start_time = 0
+			return
+		else if(prob(5))
+			to_chat(target,SPAN_DANGER(">\The [src] sucks at your lifeforce!"))
```
```suggestion
			to_chat(target,SPAN_DANGER("\The [src] sucks at your lifeforce!"))
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/starlight_structures.dm:
```diff
@@ -0,0 +1,269 @@
+	linked_god.power_per_regen += power_drain
+	. = ..()
+
+/obj/structure/deity/gateway/attack_deity(var/mob/living/deity/deity)
```
```suggestion
/obj/structure/deity/gateway/attack_deity(mob/living/deity/deity)
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/starlight_structures.dm:
```diff
@@ -0,0 +1,269 @@
+	html += "</table>"
+	show_browser(linked_god, jointext(html, null), "window=gateway")
+
+/obj/structure/deity/gateway/CanUseTopic(var/mob/user)
```
```suggestion
/obj/structure/deity/gateway/CanUseTopic(mob/user)
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/starlight_structures.dm:
```diff
@@ -0,0 +1,269 @@
+		return STATUS_INTERACTIVE
+	return STATUS_CLOSE
+
+/obj/structure/deity/gateway/proc/stop_looking_for(var/successful)
```
```suggestion
/obj/structure/deity/gateway/proc/stop_looking_for(successful)
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/starlight_structures.dm:
```diff
@@ -0,0 +1,269 @@
+			to_chat(linked_god, SPAN_WARNING("\The [src] did not find any [looking_for]. You may try again if you wish."))
+		looking_for = null
+
+/obj/structure/deity/gateway/OnTopic(var/mob/user, var/list/href_list)
```
```suggestion
/obj/structure/deity/gateway/OnTopic(mob/user, list/href_list)
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/starlight_structures.dm:
```diff
@@ -0,0 +1,269 @@
+		for(var/s in possible_forms[looking_for]["spells"])
+			var/spell/S = new s
+			H.add_spell(S)
+		GLOB.godcult.add_antagonist_mind(H.mind, 1, "[looking_for] of [linked_god]", "You are a powerful entity in the service to \the [linked_god]. [possible_forms[looking_for]["species"]]", specific_god = linked_god)
```
```suggestion
		GLOB.godcult.add_antagonist_mind(H.mind, TRUE, "[looking_for] of [linked_god]", "You are a powerful entity in the service to \the [linked_god]. [possible_forms[looking_for]["species"]]", specific_god = linked_god)
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/starlight_structures.dm:
```diff
@@ -0,0 +1,269 @@
+	power_adjustment = 1
+	deity_flags = DEITY_STRUCTURE_NEAR_IMPORTANT|DEITY_STRUCTURE_ALONE
+	var/charge = 0
+	var/charging = 0 //Charging, dispersing, etc.
```
```suggestion
	/// Charging, dispersing, etc.
	var/charging = FALSE
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/starlight_structures.dm:
```diff
@@ -0,0 +1,269 @@
+			if(get_dist(M.current, src) <= 3)
+				. += M.current
+
+/obj/structure/deity/radiant_statue/attack_hand(var/mob/living/L)
```
```suggestion
/obj/structure/deity/radiant_statue/attack_hand(mob/living/L)
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/starlight_structures.dm:
```diff
@@ -0,0 +1,269 @@
+	else
+		to_chat(L, SPAN_WARNING("\The [src] does not recognize you as a herald of \the [linked_god]. You must wear a full set of herald's armor."))
+
+/obj/structure/deity/radiant_statue/attack_deity(var/mob/living/deity/deity)
```
```suggestion
/obj/structure/deity/radiant_statue/attack_deity(mob/living/deity/deity)
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/starlight_structures.dm:
```diff
@@ -0,0 +1,269 @@
+/obj/structure/deity/radiant_statue/proc/activate_charging()
+	var/list/followers = get_followers_nearby()
+	if(is_processing || !followers.len)
+		return 0
```
```suggestion
		return FALSE
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/starlight_structures.dm:
```diff
@@ -0,0 +1,269 @@
+	var/list/followers = get_followers_nearby()
+	if(is_processing || !followers.len)
+		return 0
+	charging = 1
```
```suggestion
	charging = TRUE
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/starlight_structures.dm:
```diff
@@ -0,0 +1,269 @@
+		return 0
+	charging = 1
+	START_PROCESSING(SSobj, src)
+	src.visible_message(SPAN_NOTICE("<b>\The [src]</b> hums, activating."))
```
```suggestion
	visible_message(SPAN_NOTICE("<b>\The [src]</b> hums, activating."))
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/starlight_structures.dm:
```diff
@@ -0,0 +1,269 @@
+	START_PROCESSING(SSobj, src)
+	src.visible_message(SPAN_NOTICE("<b>\The [src]</b> hums, activating."))
+	update_icon()
+	return 1
```
```suggestion
	return TRUE
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/starlight_structures.dm:
```diff
@@ -0,0 +1,269 @@
+	update_icon()
+	return 1
+
+/obj/structure/deity/radiant_statue/attackby(var/obj/item/I, var/mob/user)
```
```suggestion
/obj/structure/deity/radiant_statue/attackby(obj/item/I, mob/user)
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/starlight_structures.dm:
```diff
@@ -0,0 +1,269 @@
+		return
+	..()
+
+/obj/structure/deity/radiant_statue/proc/charge_item(var/obj/item/I, var/mob/user)
```
```suggestion
/obj/structure/deity/radiant_statue/proc/charge_item(obj/item/I, mob/user)
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/starlight_structures.dm:
```diff
@@ -0,0 +1,269 @@
+	..()
+
+/obj/structure/deity/radiant_statue/proc/charge_item(var/obj/item/I, var/mob/user)
+	. = 0
```
```suggestion
	. = FALSE
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/starlight_structures.dm:
```diff
@@ -0,0 +1,269 @@
+		var/obj/item/gun/energy/energy = I
+		if(energy.power_supply)
+			energy.power_supply.give(energy.charge_cost * energy.max_shots)
+			. = 1
```
```suggestion
			. = TRUE
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/starlight_structures.dm:
```diff
@@ -0,0 +1,269 @@
+	else if(istype(I ,/obj/item/material/knife/ritual/shadow))
+		var/obj/item/material/knife/ritual/shadow/shad = I
+		shad.charge = initial(shad.charge)
+		. = 1
```
```suggestion
		. = TRUE
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/starlight_structures.dm:
```diff
@@ -0,0 +1,269 @@
+			return
+
+		if(charge == 40)
+			src.visible_message(SPAN_NOTICE("<b>\The [src]</b> lights up, pulsing with energy."))
```
```suggestion
			visible_message(SPAN_NOTICE("<b>\The [src]</b> lights up, pulsing with energy."))
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/starlight_structures.dm:
```diff
@@ -0,0 +1,269 @@
+
+		if(charge == 40)
+			src.visible_message(SPAN_NOTICE("<b>\The [src]</b> lights up, pulsing with energy."))
+			charging = 0
```
```suggestion
			charging = FALSE
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/starlight_structures.dm:
```diff
@@ -0,0 +1,269 @@
+
+/obj/structure/deity/radiant_statue/proc/stop_charging()
+	STOP_PROCESSING(SSobj, src)
+	src.visible_message(SPAN_NOTICE("<b>\The [src]</b> powers down, returning to it's dormant form."))
```
```suggestion
	visible_message(SPAN_NOTICE("<b>\The [src]</b> powers down, returning to it's dormant form."))
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/starlight_structures.dm:
```diff
@@ -0,0 +1,269 @@
+/obj/structure/deity/radiant_statue/proc/stop_charging()
+	STOP_PROCESSING(SSobj, src)
+	src.visible_message(SPAN_NOTICE("<b>\The [src]</b> powers down, returning to it's dormant form."))
+	charging = 0
```
```suggestion
	charging = FALSE
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/wizard_structures.dm:
```diff
@@ -0,0 +1,24 @@
+	build_cost = 700
+	health_max = 75
+
+/obj/structure/deity/wizard_recharger/attack_hand(var/mob/living/hitter)
```
```suggestion
/obj/structure/deity/wizard_recharger/attack_hand(mob/living/hitter)
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/wizard_structures.dm:
```diff
@@ -0,0 +1,24 @@
+	health_max = 75
+
+/obj/structure/deity/wizard_recharger/attack_hand(var/mob/living/hitter)
+	if(!hitter.mind || !hitter.mind.learned_spells || !hitter.mind.learned_spells.len)
```
```suggestion
	if(!hitter.mind?.learned_spells?.len)
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/wizard_structures.dm:
```diff
@@ -0,0 +1,24 @@
+		to_chat(hitter, SPAN_WARNING("You don't feel as if this will do anything for you."))
+		return
+
+	hitter.visible_message(SPAN_NOTICE("\The [hitter] dips their hands into \the [src], a soft glow emanating from them."))
```
```suggestion
	hitter.visible_message(
		SPAN_NOTICE("\The [hitter] dips their hands into \the [src], a soft glow emanating from them."),
		SPAN_NOTICE("You dip your hands into \the [src], a soft glow emanating from them.")
	)
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/wizard_structures.dm:
```diff
@@ -0,0 +1,24 @@
+		return
+
+	hitter.visible_message(SPAN_NOTICE("\The [hitter] dips their hands into \the [src], a soft glow emanating from them."))
+	if(do_after(hitter, 300, src, DO_DEFAULT | DO_TARGET_UNIQUE_ACT))
```
```suggestion
	if(do_after(hitter, 30 SECONDS, src, DO_DEFAULT | DO_TARGET_UNIQUE_ACT))
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/wizard_structures.dm:
```diff
@@ -0,0 +1,24 @@
+	hitter.visible_message(SPAN_NOTICE("\The [hitter] dips their hands into \the [src], a soft glow emanating from them."))
+	if(do_after(hitter, 300, src, DO_DEFAULT | DO_TARGET_UNIQUE_ACT))
+		for(var/s in hitter.mind.learned_spells)
+			var/spell/spell = s
```
```suggestion
		for(var/spell/spell as anything in hitter.mind.learned_spells)
```

SierraKomodo commented on code/game/gamemodes/godmode/form_items/wizard_structures.dm:
```diff
@@ -0,0 +1,24 @@
+			spell.charge_counter = spell.charge_max
+		to_chat(hitter, SPAN_NOTICE("You feel refreshed!"))
+		return
+	to_chat(hitter,SPAN_WARNING("You need to keep in contact with \the [src]!"))
```
`do_after()` already displays feedback messages on failure.

```suggestion
```

SierraKomodo commented on code/game/gamemodes/godmode/god_altar.dm:
```diff
@@ -0,0 +1,97 @@
+		to_chat(src, SPAN_DANGER("You've lost an altar!"))
+	return ..()
+
+/obj/structure/deity/altar/attackby(var/obj/item/I, var/mob/user)
```
```suggestion
/obj/structure/deity/altar/attackby(obj/item/I, mob/user)
```

SierraKomodo commented on code/game/gamemodes/godmode/god_altar.dm:
```diff
@@ -0,0 +1,97 @@
+	next_cycle = world.time + 10 SECONDS
+	cycles_before_converted--
+	if(!cycles_before_converted)
+		src.visible_message("For one thundering moment, \the [target] cries out in pain before going limp and broken.")
```
```suggestion
		visible_message("For one thundering moment, \the [target] cries out in pain before going limp and broken.")
```

SierraKomodo commented on code/game/gamemodes/godmode/god_altar.dm:
```diff
@@ -0,0 +1,97 @@
+
+
+//Used for force conversion.
+/obj/structure/deity/altar/proc/set_target(var/mob/living/L)
```
```suggestion
/obj/structure/deity/altar/proc/set_target(mob/living/L)
```

SierraKomodo commented on code/game/gamemodes/godmode/god_altar.dm:
```diff
@@ -0,0 +1,97 @@
+		if(3)
+			text = "You feel like your thought are being overriden..."
+		if(2)
+			text = "You can't.... concentrate.. must... resist!"
```
```suggestion
			text = "You can't... concentrate... must... resist!"
```

SierraKomodo commented on code/game/gamemodes/godmode/god_altar.dm:
```diff
@@ -0,0 +1,97 @@
+		if(2)
+			text = "You can't.... concentrate.. must... resist!"
+		if(1)
+			text = "Can't... resist. ... anymore."
```
```suggestion
			text = "Can't... resist... anymore."
```

SierraKomodo commented on code/game/gamemodes/godmode/god_altar.dm:
```diff
@@ -0,0 +1,97 @@
+	target = null
+	update_icon()
+
+/obj/structure/deity/altar/OnTopic(var/user, var/list/href_list)
```
```suggestion
/obj/structure/deity/altar/OnTopic(user, list/href_list)
```

SierraKomodo commented on code/game/gamemodes/godmode/god_altar.dm:
```diff
@@ -0,0 +1,97 @@
+			return TOPIC_HANDLED
+
+		M.last_special = world.time + 10 SECONDS
+		M.visible_message(SPAN_WARNING("\The [M] writhes on top of \the [src]!"), SPAN_NOTICE("You struggle against the intruding thoughts, keeping them at bay!"))
```
```suggestion
		M.visible_message(
			SPAN_WARNING("\The [M] writhes on top of \the [src]!"),
			SPAN_NOTICE("You struggle against the intruding thoughts, keeping them at bay!")
		)
```

SierraKomodo commented on code/game/gamemodes/godmode/god_pylon.dm:
```diff
@@ -0,0 +1,74 @@
+	health_max = 50
+	var/list/intuned = list()
+
+/obj/structure/deity/pylon/attack_deity(var/mob/living/deity/D)
```
```suggestion
/obj/structure/deity/pylon/attack_deity(mob/living/deity/D)
```

SierraKomodo commented on code/game/gamemodes/godmode/god_pylon.dm:
```diff
@@ -0,0 +1,74 @@
+		linked_god.leave_pylon()
+	return ..()
+
+/obj/structure/deity/pylon/attack_hand(var/mob/living/L)
```
```suggestion
/obj/structure/deity/pylon/attack_hand(mob/living/L)
```

SierraKomodo commented on code/game/gamemodes/godmode/god_pylon.dm:
```diff
@@ -0,0 +1,74 @@
+	else
+		add_intuned(L)
+
+/obj/structure/deity/pylon/proc/add_intuned(var/mob/living/L)
```
```suggestion
/obj/structure/deity/pylon/proc/add_intuned(mob/living/L)
```

SierraKomodo commented on code/game/gamemodes/godmode/god_pylon.dm:
```diff
@@ -0,0 +1,74 @@
+	intuned += L
+	GLOB.destroyed_event.register(L,src,/obj/structure/deity/pylon/proc/remove_intuned)
+
+/obj/structure/deity/pylon/proc/remove_intuned(var/mob/living/L)
```
```suggestion
/obj/structure/deity/pylon/proc/remove_intuned(mob/living/L)
```

SierraKomodo commented on code/game/gamemodes/godmode/god_pylon.dm:
```diff
@@ -0,0 +1,74 @@
+	intuned -= L
+	GLOB.destroyed_event.unregister(L, src)
+
+/obj/structure/deity/pylon/OnTopic(var/mob/living/carbon/human/user, var/href_list)
```
```suggestion
/obj/structure/deity/pylon/OnTopic(mob/living/carbon/human/user, href_list)
```

SierraKomodo commented on code/game/gamemodes/godmode/god_pylon.dm:
```diff
@@ -0,0 +1,74 @@
+		. = TOPIC_REFRESH
+	. = ..()
+
+/obj/structure/deity/pylon/hear_talk(mob/M as mob, text, verb, datum/language/speaking)
```
```suggestion
/obj/structure/deity/pylon/hear_talk(mob/M, text, verb, datum/language/speaking)
```

SierraKomodo commented on code/game/gamemodes/godmode/god_structures.dm:
```diff
@@ -0,0 +1,60 @@
+/proc/valid_deity_structure_spot(var/type, var/turf/target, var/mob/living/deity/deity, var/mob/living/user)
```
```suggestion
/proc/valid_deity_structure_spot(type, turf/target, mob/living/deity/deity, mob/living/user)
```

SierraKomodo commented on code/game/gamemodes/godmode/god_structures.dm:
```diff
@@ -0,0 +1,60 @@
+	if(flags & DEITY_STRUCTURE_NEAR_IMPORTANT && !deity.near_structure(target))
+		if(user)
+			to_chat(user, SPAN_WARNING("You need to be near \a [deity.get_type_name(/obj/structure/deity/altar)] to build this!"))
+		return 0
```
```suggestion
		return FALSE
```

SierraKomodo commented on code/game/gamemodes/godmode/god_structures.dm:
```diff
@@ -0,0 +1,60 @@
+			if(istype(structure,type) && get_dist(target,structure) <= 3)
+				if(user)
+					to_chat(user, SPAN_WARNING("You are too close to another [deity.get_type_name(type)]!"))
+				return 0
```
```suggestion
				return FALSE
```

SierraKomodo commented on code/game/gamemodes/godmode/god_structures.dm:
```diff
@@ -0,0 +1,60 @@
+				if(user)
+					to_chat(user, SPAN_WARNING("You are too close to another [deity.get_type_name(type)]!"))
+				return 0
+	return 1
```
```suggestion
	return TRUE
```

SierraKomodo commented on code/game/gamemodes/godmode/god_structures.dm:
```diff
@@ -0,0 +1,60 @@
+		DAMAGE_OXY       = 0,
+		DAMAGE_BRAIN     = 0
+	)
+	var/power_adjustment = 1 //How much power we get/lose
```
```suggestion
	/// How much power we get/lose
	var/power_adjustment = 1
```

SierraKomodo commented on code/game/gamemodes/godmode/god_structures.dm:
```diff
@@ -0,0 +1,60 @@
+		DAMAGE_BRAIN     = 0
+	)
+	var/power_adjustment = 1 //How much power we get/lose
+	var/build_cost = 0 //How much it costs to build this item.
```
```suggestion
	/// How much it costs to build this item.
	var/build_cost = 0
```

SierraKomodo commented on code/game/gamemodes/godmode/god_structures.dm:
```diff
@@ -0,0 +1,60 @@
+	anchored = TRUE
+	icon_state = "tomealtar"
+
+/obj/structure/deity/New(var/newloc, var/god)
```
```suggestion
/obj/structure/deity/New(newloc, god)
```

SierraKomodo commented on code/game/gamemodes/godmode/god_structures.dm:
```diff
@@ -0,0 +1,60 @@
+/obj/structure/deity/post_health_change(health_mod, damage_type)
+	update_icon()
+
+/obj/structure/deity/proc/attack_deity(var/mob/living/deity/deity)
```
```suggestion
/obj/structure/deity/proc/attack_deity(mob/living/deity/deity)
```

SierraKomodo commented on code/game/gamemodes/godmode/god_trap.dm:
```diff
@@ -0,0 +1,30 @@
+	. = ..()
+	GLOB.entered_event.register(get_turf(src), src, /obj/structure/deity/trap/proc/trigger)
+
+/obj/structure/deity/trap/attackby(obj/item/W as obj, mob/user as mob)
```
```suggestion
/obj/structure/deity/trap/attackby(obj/item/W, mob/user)
```

SierraKomodo commented on code/game/gamemodes/godmode/god_trap.dm:
```diff
@@ -0,0 +1,30 @@
+/obj/structure/deity/trap/bullet_act()
+	return
+
+/obj/structure/deity/trap/proc/trigger(var/atom/entered, var/atom/movable/enterer)
```
```suggestion
/obj/structure/deity/trap/proc/trigger(atom/entered, atom/movable/enterer)
```

SierraKomodo commented on code/game/objects/items/weapons/weaponry.dm:
```diff
@@ -45,6 +45,28 @@
 
 	..()
 
+/obj/item/weapon/nullrod/afterattack(var/atom/A, var/mob/user, var/proximity)
```
```suggestion
/obj/item/weapon/nullrod/afterattack(atom/A, mob/user, proximity)
```

SierraKomodo commented on code/game/objects/items/weapons/weaponry.dm:
```diff
@@ -45,6 +45,28 @@
+		if(!altar.linked_god.silenced) //Don't want them to infinity spam it.
+			altar.linked_god.silence(10)
+			new /obj/effect/temporary(get_turf(altar),'icons/effects/effects.dmi',"purple_electricity_constant", 10)
+			altar.visible_message("<span class='notice'>\The [altar] groans in protest as reality settles around \the [src].</span>")
```
```suggestion
			altar.visible_message(SPAN_NOTICE("\The [altar] groans in protest as reality settles around \the [src]."))
```

SierraKomodo commented on code/game/objects/items/weapons/weaponry.dm:
```diff
@@ -45,6 +45,28 @@
+
+	if(istype(A, /turf/simulated/wall/cult))
+		var/turf/simulated/wall/cult/W = A
+		user.visible_message("<span class='notice'>\The [user] touches \the [A] with \the [src], and the enchantment affecting it fizzles away.</span>", "<span class='notice'>You touch \the [A] with \the [src], and the enchantment affecting it fizzles away.</span>")
```
```suggestion
		user.visible_message(
			SPAN_NOTICE("\The [user] touches \the [A] with \the [src], and the enchantment affecting it fizzles away."),
			SPAN_NOTICE("You touch \the [A] with \the [src], and the enchantment affecting it fizzles away.")
		)
```

SierraKomodo commented on code/game/objects/items/weapons/weaponry.dm:
```diff
@@ -45,6 +45,28 @@
+
+	if(istype(A, /turf/simulated/floor/cult))
+		var/turf/simulated/floor/cult/F = A
+		user.visible_message("<span class='notice'>\The [user] touches \the [A] with \the [src], and the enchantment affecting it fizzles away.</span>", "<span class='notice'>You touch \the [A] with \the [src], and the enchantment affecting it fizzles away.</span>")
```
```suggestion
		user.visible_message(
			SPAN_NOTICE("\The [user] touches \the [A] with \the [src], and the enchantment affecting it fizzles away."),
			SPAN_NOTICE("You touch \the [A] with \the [src], and the enchantment affecting it fizzles away.")
		)
```

SierraKomodo commented on code/game/turfs/flooring/flooring.dm:
```diff
@@ -384,11 +384,11 @@
 	icon_base = "flesh"
 	descriptor = "flesh"
+	remove_timer = 60
 	has_base_range = 3
 	damage_temperature = T0C + 100
 	build_type = null
 	flags = TURF_ACID_IMMUNE | TURF_REMOVE_CROWBAR
-	remove_timer = 60
 	can_engrave = FALSE
```
Wild edit that doesn't actually change anything

SierraKomodo commented on code/modules/mob/living/deity/deity.dm:
```diff
@@ -0,0 +1,119 @@
+	bone_amount = 0
+
+	var/eye_type = /mob/observer/eye/cult
+	var/list/minions = list() //Minds of those who follow him
```
```suggestion
	/// Minds of those who follow him
	var/list/minions = list()
```

SierraKomodo commented on code/modules/mob/living/deity/deity.dm:
```diff
@@ -0,0 +1,119 @@
+
+	var/eye_type = /mob/observer/eye/cult
+	var/list/minions = list() //Minds of those who follow him
+	var/list/structures = list() //The objs that this dude controls.
```
```suggestion
	/// The objs that this dude controls.
	var/list/structures = list()
```

SierraKomodo commented on code/modules/mob/living/deity/deity.dm:
```diff
@@ -0,0 +1,119 @@
+	return STATUS_INTERACTIVE
+
+/mob/living/deity/Destroy()
+	death(0)
```
0/FALSE is unnecessary here - That parameter already defaults as false.

```suggestion
	death()
```

SierraKomodo commented on code/modules/mob/living/deity/deity.dm:
```diff
@@ -0,0 +1,119 @@
+	dat += "</table>"
+	show_browser(src, JOINTEXT(dat), "window=godform;can_close=0")
+
+/mob/living/deity/proc/set_form(var/type)
```
```suggestion
/mob/living/deity/proc/set_form(type)
```

SierraKomodo commented on code/modules/mob/living/deity/deity.dm:
```diff
@@ -0,0 +1,119 @@
+	form = new type(src)
+	to_chat(src, SPAN_NOTICE("You undergo a transformation into your new form!"))
+	addtimer(CALLBACK(src, .proc/GodName), 1 SECONDS)
+	src.verbs -= /mob/living/deity/verb/choose_form
```
```suggestion
	verbs -= /mob/living/deity/verb/choose_form
```

SierraKomodo commented on code/modules/mob/living/deity/deity.dm:
```diff
@@ -0,0 +1,119 @@
+		var/mob/living/L = mind.current
+		L.faction = form.faction
+
+/mob/living/deity/proc/GodName(var/newname)
```
```suggestion
/mob/living/deity/proc/GodName(newname)
```

SierraKomodo commented on code/modules/mob/living/deity/deity.dm:
```diff
@@ -0,0 +1,119 @@
+	if(newname)
+		fully_replace_character_name(newname)
+
+//Gets the name based on form, or if there is no form name, type.
```
```suggestion
/// Gets the name based on form, or if there is no form name, type.
```

SierraKomodo commented on code/modules/mob/living/deity/deity.dm:
```diff
@@ -0,0 +1,119 @@
+		fully_replace_character_name(newname)
+
+//Gets the name based on form, or if there is no form name, type.
+/mob/living/deity/proc/get_type_name(var/type)
```
```suggestion
/mob/living/deity/proc/get_type_name(type)
```

SierraKomodo commented on code/modules/mob/living/deity/deity_boons.dm:
```diff
@@ -0,0 +1,55 @@
+/mob/living/deity/proc/set_boon(var/datum/boon)
```
```suggestion
/mob/living/deity/proc/set_boon(datum/boon)
```

SierraKomodo commented on code/modules/mob/living/deity/deity_boons.dm:
```diff
@@ -0,0 +1,55 @@
+		var/spell/S = boon
+		nano_data["boon_name"] = S.name
+
+/mob/living/deity/proc/grant_boon(var/mob/living/L)
```
```suggestion
/mob/living/deity/proc/grant_boon(mob/living/L)
```

SierraKomodo commented on code/modules/mob/living/deity/deity_boons.dm:
```diff
@@ -0,0 +1,55 @@
+	nano_data["boon_name"] = null
+	return
+
+/mob/living/deity/proc/grant_spell(var/mob/living/target, var/spell/spell)
```
```suggestion
/mob/living/deity/proc/grant_spell(mob/living/target, spell/spell)
```

SierraKomodo commented on code/modules/mob/living/deity/deity_boons.dm:
```diff
@@ -0,0 +1,55 @@
+	for(var/s in M.learned_spells)
+		var/spell/S = s
+		if(istype(S, spell.type))
+			to_chat(src, SPAN_WARNING("They already know that spell!"))
```
```suggestion
			to_chat(src, SPAN_WARNING("\The [target] already knows that spell!"))
```

SierraKomodo commented on code/modules/mob/living/deity/deity_boons.dm:
```diff
@@ -0,0 +1,55 @@
+		var/spell/S = s
+		if(istype(S, spell.type))
+			to_chat(src, SPAN_WARNING("They already know that spell!"))
+			return 0
```
```suggestion
			return FALSE
```

SierraKomodo commented on code/modules/mob/living/deity/deity_boons.dm:
```diff
@@ -0,0 +1,55 @@
+	target.add_spell(spell)
+	spell.set_connected_god(src)
+	to_chat(target, SPAN_NOTICE("You feel a surge of power as you learn the art of [current_boon]."))
+	return 1
```
```suggestion
	return TRUE
```

SierraKomodo commented on code/modules/mob/living/deity/deity_boons.dm:
```diff
@@ -0,0 +1,55 @@
+	return 1
+
+/* This is a generic proc used by the God to inact a sacrifice from somebody. Power is a value of magnitude.
+*/
```
```suggestion
/// This is a generic proc used by the God to inact a sacrifice from somebody. Power is a value of magnitude.
```

SierraKomodo commented on code/modules/mob/living/deity/deity_boons.dm:
```diff
@@ -0,0 +1,55 @@
+
+/* This is a generic proc used by the God to inact a sacrifice from somebody. Power is a value of magnitude.
+*/
+/mob/living/deity/proc/take_charge(var/mob/living/L, var/power)
```
```suggestion
/mob/living/deity/proc/take_charge(mob/living/L, power)
```

SierraKomodo commented on code/modules/mob/living/deity/deity_boons.dm:
```diff
@@ -0,0 +1,55 @@
+/mob/living/deity/proc/take_charge(var/mob/living/L, var/power)
+	if(form)
+		return form.take_charge(L, power)
+	return 1
```
```suggestion
	return TRUE
```

SierraKomodo commented on code/modules/mob/living/deity/deity_click.dm:
```diff
@@ -0,0 +1,24 @@
+/mob/living/deity/ClickOn(var/atom/A, var/params)
```
```suggestion
/mob/living/deity/ClickOn(atom/A, params)
```

SierraKomodo commented on code/modules/mob/living/deity/deity_items.dm:
```diff
@@ -0,0 +1,44 @@
+/mob/living/deity
+	var/list/items
+	var/list/items_by_category
```
This prevents VSC from jumping to this file when trying to find the original definition of `/mob/living/deity`

```suggestion
/mob/living/deity/var/list/items
/mob/living/deity/var/list/items_by_category
```

SierraKomodo commented on code/modules/mob/living/deity/deity_items.dm:
```diff
@@ -0,0 +1,44 @@
+	var/list/items
+	var/list/items_by_category
+
+/mob/living/deity/proc/set_items(var/list/_items)
```
```suggestion
/mob/living/deity/proc/set_items(list/_items)
```

SierraKomodo commented on code/modules/mob/living/deity/deity_items.dm:
```diff
@@ -0,0 +1,44 @@
+			items_by_category[di.category] = list()
+		items_by_category[di.category] += di
+
+/mob/living/deity/proc/has_item(var/name, var/minimum_level = 1)
```
```suggestion
/mob/living/deity/proc/has_item(name, minimum_level = 1)
```

SierraKomodo commented on code/modules/mob/living/deity/deity_items.dm:
```diff
@@ -0,0 +1,44 @@
+	var/datum/deity_item/di = items[name]
+	. = di.level >= minimum_level
+
+/mob/living/deity/proc/upgrade_item(var/name)
```
```suggestion
/mob/living/deity/proc/upgrade_item(name)
```

SierraKomodo commented on code/modules/mob/living/deity/deity_items.dm:
```diff
@@ -0,0 +1,44 @@
+	di.buy(src)
+	. = TRUE
+
+/mob/living/deity/proc/get_item_level(var/name)
```
```suggestion
/mob/living/deity/proc/get_item_level(name)
```

SierraKomodo commented on code/modules/mob/living/deity/deity_phenomena.dm:
```diff
@@ -0,0 +1,95 @@
+	var/silenced = 0
+	var/list/phenomenas = list()
+	var/list/intent_phenomenas = list()
+	var/static/list/control_types = list("control", "controlshift", "shift")
```
```suggestion
/mob/living/deity/var/silenced = FALSE
/mob/living/deity/var/list/phenomenas = list()
/mob/living/deity/var/list/intent_phenomenas = list()
/mob/living/deity/var/static/list/control_types = list("control", "controlshift", "shift")
```

SierraKomodo commented on code/modules/mob/living/deity/deity_phenomena.dm:
```diff
@@ -0,0 +1,95 @@
+	set_phenomena(add_phenomena(/datum/phenomena/conversion), I_GRAB, "shift")
+	set_phenomena(add_phenomena(/datum/phenomena/forced_conversion), I_GRAB, "control")
+
+/mob/living/deity/proc/silence(var/amount)
```
```suggestion
/mob/living/deity/proc/silence(amount)
```

SierraKomodo commented on code/modules/mob/living/deity/deity_phenomena.dm:
```diff
@@ -0,0 +1,95 @@
+		remove_phenomena(phenom)
+	return ..()
+
+/mob/living/deity/proc/add_phenomena(var/type)
```
```suggestion
/mob/living/deity/proc/add_phenomena(type)
```

SierraKomodo commented on code/modules/mob/living/deity/deity_phenomena.dm:
```diff
@@ -0,0 +1,95 @@
+	phenomenas[P.name] = P
+	return P
+
+/mob/living/deity/proc/remove_phenomena_from_intent(var/intent, var/modifier, var/update = 1)
```
```suggestion
/mob/living/deity/proc/remove_phenomena_from_intent(intent, modifier, update = TRUE)
```

SierraKomodo commented on code/modules/mob/living/deity/deity_phenomena.dm:
```diff
@@ -0,0 +1,95 @@
+	if(update)
+		update_phenomena_bindings()
+
+/mob/living/deity/proc/remove_phenomena(var/to_remove)
```
```suggestion
/mob/living/deity/proc/remove_phenomena(to_remove)
```

SierraKomodo commented on code/modules/mob/living/deity/deity_phenomena.dm:
```diff
@@ -0,0 +1,95 @@
+		selected = null
+	qdel(P)
+
+/mob/living/deity/proc/populate_intent(var/intent)
```
```suggestion
/mob/living/deity/proc/populate_intent(intent)
```

SierraKomodo commented on code/modules/mob/living/deity/deity_phenomena.dm:
```diff
@@ -0,0 +1,95 @@
+		intent_phenomenas[intent] = list()
+	intent_phenomenas[intent] |= control_types
+
+/mob/living/deity/proc/set_phenomena(var/datum/phenomena/phenomena, var/intent, var/modifiers)
```
```suggestion
/mob/living/deity/proc/set_phenomena(datum/phenomena/phenomena, intent, modifiers)
```

SierraKomodo commented on code/modules/mob/living/deity/deity_phenomena.dm:
```diff
@@ -0,0 +1,95 @@
+	var/list/intent_list = intent_phenomenas[intent]
+	intent_list[modifiers] = phenomena
+
+/mob/living/deity/proc/get_phenomena(var/shift = 0, var/control = 0)
```
```suggestion
/mob/living/deity/proc/get_phenomena(shift = FALSE, control = FALSE)
```

SierraKomodo commented on code/modules/mob/living/deity/deity_power.dm:
```diff
@@ -0,0 +1,26 @@
+		if(power_per_regen < 0 || power < power_min)
+			adjust_power(power_per_regen)
+
+/mob/living/deity/proc/adjust_power(var/amount)
```
```suggestion
/mob/living/deity/proc/adjust_power(amount)
```

SierraKomodo commented on code/modules/mob/living/deity/deity_power.dm:
```diff
@@ -0,0 +1,26 @@
+	if(amount)
+		power = max(0, power + amount)
+
+/mob/living/deity/proc/adjust_power_min(var/amount, var/silent = 0, var/msg)
```
```suggestion
/mob/living/deity/proc/adjust_power_min(amount, silent = FALSE, msg)
```

SierraKomodo commented on code/modules/mob/living/deity/deity_pylon.dm:
```diff
@@ -0,0 +1,21 @@
+	var/image/pylon_image
+	var/obj/structure/deity/pylon/pylon
+
+/mob/living/deity/set_form(var/type)
```
```suggestion
/mob/living/deity/set_form(type)
```

SierraKomodo commented on code/modules/mob/living/deity/deity_pylon.dm:
```diff
@@ -0,0 +1,21 @@
+	pylon_image = image('icons/mob/mob.dmi', icon_state = form.pylon_icon_state)
+	pylon_image.alpha = 180
+
+/mob/living/deity/proc/possess_pylon(var/obj/structure/deity/pylon/P)
```
```suggestion
/mob/living/deity/proc/possess_pylon(obj/structure/deity/pylon/P)
```

SierraKomodo commented on code/modules/mob/living/deity/deity_sources.dm:
```diff
@@ -0,0 +1,84 @@
+/mob/living/deity/proc/add_follower(var/mob/living/L)
```
```suggestion
/mob/living/deity/proc/add_follower(mob/living/L)
```

SierraKomodo commented on code/modules/mob/living/deity/deity_sources.dm:
```diff
@@ -0,0 +1,84 @@
+/mob/living/deity/proc/add_follower(var/mob/living/L)
+	if(is_follower(L, silent=1))
```
```suggestion
	if(is_follower(L, TRUE))
```

SierraKomodo commented on code/modules/mob/living/deity/deity_sources.dm:
```diff
@@ -0,0 +1,84 @@
+	GLOB.destroyed_event.register(L,src, .proc/dead_follower)
+	GLOB.death_event.register(L,src, .proc/update_followers)
+
+/mob/living/deity/proc/dead_follower(var/mob/living/L)
```
```suggestion
/mob/living/deity/proc/dead_follower(mob/living/L)
```

SierraKomodo commented on code/modules/mob/living/deity/deity_sources.dm:
```diff
@@ -0,0 +1,84 @@
+	GLOB.death_event.unregister(L,src)
+	GLOB.destroyed_event.unregister(L,src)
+
+/mob/living/deity/proc/remove_follower_spells(var/datum/mind/M)
```
```suggestion
/mob/living/deity/proc/remove_follower_spells(datum/mind/M)
```

SierraKomodo commented on code/modules/mob/living/deity/deity_sources.dm:
```diff
@@ -0,0 +1,84 @@
+				M.current.remove_spell(S)
+				qdel(S)
+
+/mob/living/deity/proc/remove_follower(var/mob/living/L)
```
```suggestion
/mob/living/deity/proc/remove_follower(mob/living/L)
```

SierraKomodo commented on code/modules/mob/living/deity/deity_sources.dm:
```diff
@@ -0,0 +1,84 @@
+				qdel(S)
+
+/mob/living/deity/proc/remove_follower(var/mob/living/L)
+	if(!is_follower(L, silent=1))
```
```suggestion
	if(!is_follower(L, TRUE))
```

SierraKomodo commented on code/modules/mob/living/deity/deity_sources.dm:
```diff
@@ -0,0 +1,84 @@
+	update_followers()
+
+
+/mob/living/deity/proc/adjust_source(var/amount, var/atom/source, var/silent = 0, var/msg)
```
```suggestion
/mob/living/deity/proc/adjust_source(amount, atom/source, silent = FALSE, msg)
```

SierraKomodo commented on code/modules/mob/living/deity/deity_sources.dm:
```diff
@@ -0,0 +1,84 @@
+		if(istype(source, /obj/structure/deity))
+			structures -= source
+
+/mob/living/deity/proc/is_follower(var/mob/living/L, var/silent = 0)
```
```suggestion
/mob/living/deity/proc/is_follower(mob/living/L, silent = FALSE)
```

SierraKomodo commented on code/modules/mob/living/deity/deity_sources.dm:
```diff
@@ -0,0 +1,84 @@
+	if(istype(L))
+		if(L.mind)
+			if(L.mind in minions)
+				return 1
```
```suggestion
				return TRUE
```

SierraKomodo commented on code/modules/mob/living/deity/deity_sources.dm:
```diff
@@ -0,0 +1,84 @@
+				return 1
+		if(!silent)
+			to_chat(src, SPAN_WARNING("You do not feel a malleable mind behind that frame."))
+	return 0
```
```suggestion
	return FALSE
```

SierraKomodo commented on code/modules/mob/living/deity/deity_sources.dm:
```diff
@@ -0,0 +1,84 @@
+			to_chat(src, SPAN_WARNING("You do not feel a malleable mind behind that frame."))
+	return 0
+
+/mob/living/deity/fully_replace_character_name(var/new_name, var/in_depth = TRUE)
```
```suggestion
/mob/living/deity/fully_replace_character_name(new_name, in_depth = TRUE)
```

SierraKomodo commented on code/modules/mob/living/deity/deity_sources.dm:
```diff
@@ -0,0 +1,84 @@
+
+/mob/living/deity/fully_replace_character_name(var/new_name, var/in_depth = TRUE)
+	if(!..())
+		return 0
```
```suggestion
		return FALSE
```

SierraKomodo commented on code/modules/mob/living/deity/deity_sources.dm:
```diff
@@ -0,0 +1,84 @@
+		minion.special_role = "Servant of [new_name]"
+	eyeobj.SetName("[src] ([eyeobj.name_sufix])")
+	nano_data["name"] = new_name
+	return 1
```
```suggestion
	return TRUE
```

SierraKomodo commented on code/modules/mob/living/deity/deity_sources.dm:
```diff
@@ -0,0 +1,84 @@
+	nano_data["name"] = new_name
+	return 1
+
+//Whether we are near an important structure.
```
```suggestion
/// Whether we are near an important structure.
```

SierraKomodo commented on code/modules/mob/living/deity/deity_sources.dm:
```diff
@@ -0,0 +1,84 @@
+	return 1
+
+//Whether we are near an important structure.
+/mob/living/deity/proc/near_structure(var/atom/A, var/all_structures = 0)
```
```suggestion
/mob/living/deity/proc/near_structure(atom/A, all_structures = FALSE)
```

SierraKomodo commented on code/modules/mob/living/deity/deity_sources.dm:
```diff
@@ -0,0 +1,84 @@
+				continue
+
+		if(get_dist(T, s) <= 3)
+			return 1
```
```suggestion
			return TRUE
```

SierraKomodo commented on code/modules/mob/living/deity/deity_sources.dm:
```diff
@@ -0,0 +1,84 @@
+
+		if(get_dist(T, s) <= 3)
+			return 1
+	return 0
```
```suggestion
	return FALSE
```

SierraKomodo commented on code/modules/mob/living/deity/deity_topic.dm:
```diff
@@ -0,0 +1,51 @@
+			set_form(type)
+		return TOPIC_HANDLED
+	if(href_list["select_phenomena"])
+		nano_data["phenomenaMenu"] = 1
```
```suggestion
		nano_data["phenomenaMenu"] = TRUE
```

SierraKomodo commented on code/modules/mob/living/deity/deity_topic.dm:
```diff
@@ -0,0 +1,51 @@
+		nano_data["selectedPhenomenaName"] = selected.name
+		return TOPIC_HANDLED
+	if(href_list["clear_selected"])
+		nano_data["phenomenaMenu"] = 0
```
```suggestion
		nano_data["phenomenaMenu"] = FALSE
```

SierraKomodo commented on code/modules/mob/living/deity/deity_topic.dm:
```diff
@@ -0,0 +1,51 @@
+		return TOPIC_HANDLED
+	if(href_list["jump"])
+		var/atom/a = locate(href_list["jump"])
+		var/follow = 0
```
```suggestion
		var/follow = FALSE
```

SierraKomodo commented on code/modules/mob/living/deity/deity_topic.dm:
```diff
@@ -0,0 +1,51 @@
+		var/atom/a = locate(href_list["jump"])
+		var/follow = 0
+		if(href_list["follow"])
+			follow = 1
```
```suggestion
			follow = TRUE
```

SierraKomodo commented on code/modules/mob/living/deity/deity_topic.dm:
```diff
@@ -0,0 +1,51 @@
+		return TOPIC_HANDLED
+	if(href_list["switchCategory"])
+		set_nano_category(text2num(href_list["switchCategory"]))
+		return 1
```
```suggestion
		return TOPIC_HANDLED
```

SierraKomodo commented on code/modules/mob/living/deity/deity_tracking.dm:
```diff
@@ -0,0 +1,40 @@
+		follow_follower(choice)
+
+/mob/living/deity/proc/follow_follower(var/mob/living/L)
+	if(!L || L.stat == DEAD || !is_follower(L, silent=1))
```
```suggestion
	if(!L || L.stat == DEAD || !is_follower(L, TRUE))
```

SierraKomodo commented on code/modules/mob/living/deity/deity_tracking.dm:
```diff
@@ -0,0 +1,40 @@
+	to_chat(src, SPAN_NOTICE("You stop following \the [following]."))
+	following = null
+
+/mob/living/deity/proc/keep_following(var/atom/movable/moving_instance, var/atom/old_loc, var/atom/new_loc)
```
```suggestion
/mob/living/deity/proc/keep_following(atom/movable/moving_instance, atom/old_loc, atom/new_loc)
```

SierraKomodo commented on code/modules/mob/living/deity/deity_tracking.dm:
```diff
@@ -0,0 +1,40 @@
+	if(choice)
+		follow_follower(choice)
+
+/mob/living/deity/proc/follow_follower(var/mob/living/L)
```
```suggestion
/mob/living/deity/proc/follow_follower(mob/living/L)
```

SierraKomodo commented on code/modules/mob/living/deity/forms.dm:
```diff
@@ -0,0 +1,48 @@
+	var/name = "Form"
+	var/info = "This is a form your being can take."
+	var/desc = "This is the mob's description given."
+	var/faction = MOB_FACTION_NEUTRAL //For stuff like mobs and shit
```
```suggestion
	/// For stuff like mobs and shit
	var/faction = MOB_FACTION_NEUTRAL
```

SierraKomodo commented on code/modules/mob/living/deity/forms.dm:
```diff
@@ -0,0 +1,48 @@
+	var/info = "This is a form your being can take."
+	var/desc = "This is the mob's description given."
+	var/faction = MOB_FACTION_NEUTRAL //For stuff like mobs and shit
+	var/god_icon_state = "nar-sie" //What you look like
```
```suggestion
	/// What you look like
	var/god_icon_state = "nar-sie"
```

SierraKomodo commented on code/modules/mob/living/deity/forms.dm:
```diff
@@ -0,0 +1,48 @@
+	var/desc = "This is the mob's description given."
+	var/faction = MOB_FACTION_NEUTRAL //For stuff like mobs and shit
+	var/god_icon_state = "nar-sie" //What you look like
+	var/pylon_icon_state //What image shows up when appearing in a pylon
```
```suggestion
	/// What image shows up when appearing in a pylon
	var/pylon_icon_state
```

SierraKomodo commented on code/modules/mob/living/deity/forms.dm:
```diff
@@ -0,0 +1,48 @@
+	var/mob/living/deity/linked_god = null
+	var/starting_power_min = 10
+	var/starting_regeneration = 1
+	var/list/buildables = list() //Both a list of var changes and a list of buildables.
```
```suggestion
	/// Both a list of var changes and a list of buildables.
	var/list/buildables = list()
```

SierraKomodo commented on code/modules/mob/living/deity/forms.dm:
```diff
@@ -0,0 +1,48 @@
+	var/list/icon_states = list()
+	var/list/items
+
+/datum/god_form/New(var/mob/living/deity/D)
```
```suggestion
/datum/god_form/New(mob/living/deity/D)
```

SierraKomodo commented on code/modules/mob/living/deity/forms.dm:
```diff
@@ -0,0 +1,48 @@
+		D.set_items(complete_items)
+		items.Cut()
+
+/datum/god_form/proc/sync_structure(var/obj/O)
```
```suggestion
/datum/god_form/proc/sync_structure(obj/O)
```

SierraKomodo commented on code/modules/mob/living/deity/forms.dm:
```diff
@@ -0,0 +1,48 @@
+	for(var/V in svars)
+		O.vars[V] = svars[V]
+
+/datum/god_form/proc/take_charge(var/mob/living/user, var/charge)
```
```suggestion
/datum/god_form/proc/take_charge(mob/living/user, charge)
```

SierraKomodo commented on code/modules/mob/living/deity/forms.dm:
```diff
@@ -0,0 +1,48 @@
+		O.vars[V] = svars[V]
+
+/datum/god_form/proc/take_charge(var/mob/living/user, var/charge)
+	return 1
```
```suggestion
	return TRUE
```

SierraKomodo commented on code/modules/mob/living/deity/forms/narsie.dm:
```diff
@@ -0,0 +1,49 @@
+
+	buildables = list(/obj/structure/deity/altar/narsie,
+					/obj/structure/deity/pylon
+					)
```
```suggestion
	buildables = list(
		/obj/structure/deity/altar/narsie,
		/obj/structure/deity/pylon
	)
```

SierraKomodo commented on code/modules/mob/living/deity/forms/narsie.dm:
```diff
@@ -0,0 +1,49 @@
+				/datum/deity_item/blood_crafting,
+				/datum/deity_item/blood_crafting/armored,
+				/datum/deity_item/blood_crafting/space
+				)
```
```suggestion
	items = list(
		/datum/deity_item/general/potential,
		/datum/deity_item/general/regeneration,
		/datum/deity_item/boon/eternal_darkness,
		/datum/deity_item/boon/torment,
		/datum/deity_item/boon/blood_shard,
		/datum/deity_item/boon/drain_blood,
		/datum/deity_item/phenomena/exhude_blood,
		/datum/deity_item/sacrifice,
		/datum/deity_item/boon/sac_dagger,
		/datum/deity_item/boon/sac_spell,
		/datum/deity_item/boon/execution_axe,
		/datum/deity_item/blood_stone,
		/datum/deity_item/minions,
		/datum/deity_item/boon/soul_shard,
		/datum/deity_item/boon/blood_zombie,
		/datum/deity_item/boon/tear_veil,
		/datum/deity_item/phenomena/hellscape,
		/datum/deity_item/blood_crafting,
		/datum/deity_item/blood_crafting/armored,
		/datum/deity_item/blood_crafting/space
	)
```

SierraKomodo commented on code/modules/mob/living/deity/forms/narsie.dm:
```diff
@@ -0,0 +1,49 @@
+				/datum/deity_item/blood_crafting/space
+				)
+
+/datum/god_form/narsie/take_charge(var/mob/living/user, var/charge)
```
```suggestion
/datum/god_form/narsie/take_charge(mob/living/user, charge)
```

SierraKomodo commented on code/modules/mob/living/deity/forms/narsie.dm:
```diff
@@ -0,0 +1,49 @@
+/datum/god_form/narsie/take_charge(var/mob/living/user, var/charge)
+	charge = min(100, charge * 0.25)
+	if(prob(charge))
+		to_chat(user, "<span class='warning'>You feel drained...</span>")
```
```suggestion
		to_chat(user, SPAN_WARNING("You feel drained...")
```

SierraKomodo commented on code/modules/mob/living/deity/forms/narsie.dm:
```diff
@@ -0,0 +1,49 @@
+		H.vessel.remove_reagent(/datum/reagent/blood, charge)
+	else
+		user.adjustBruteLoss(charge)
+	return 1
```
```suggestion
	return TRUE
```

SierraKomodo commented on code/modules/mob/living/deity/forms/starlight.dm:
```diff
@@ -0,0 +1,47 @@
+	buildables = list(/obj/structure/deity/altar/starlight,
+					/obj/structure/deity/pylon/starlight,
+					/obj/structure/deity/radiant_statue,
+					)
```
```suggestion
	buildables = list(
		/obj/structure/deity/altar/starlight,
		/obj/structure/deity/pylon/starlight,
		/obj/structure/deity/radiant_statue,
	)
```

SierraKomodo commented on code/modules/mob/living/deity/forms/starlight.dm:
```diff
@@ -0,0 +1,47 @@
+				/datum/deity_item/phenomena/burning_glare,
+				/datum/deity_item/phenomena/open_gateway,
+				/datum/deity_item/phenomena/divine_right
+				)
```
```suggestion
	items = list(
		/datum/deity_item/general/potential,
		/datum/deity_item/general/regeneration,
		/datum/deity_item/boon/blazing_blade,
		/datum/deity_item/boon/holy_beacon,
		/datum/deity_item/boon/black_death,
		/datum/deity_item/blood_crafting/firecrafting,
		/datum/deity_item/boon/starburst,
		/datum/deity_item/boon/exchange_wounds,
		/datum/deity_item/boon/radiant_aura,
		/datum/deity_item/boon/burning_touch,
		/datum/deity_item/boon/burning_grip,
		/datum/deity_item/boon/blood_boil,
		/datum/deity_item/boon/emp,
		/datum/deity_item/boon/cure_light,
		/datum/deity_item/phenomena/herald,
		/datum/deity_item/phenomena/wisp,
		/datum/deity_item/phenomena/flickering_whisper,
		/datum/deity_item/phenomena/burning_glare,
		/datum/deity_item/phenomena/open_gateway,
		/datum/deity_item/phenomena/divine_right
	)
```

SierraKomodo commented on code/modules/mob/living/deity/forms/starlight.dm:
```diff
@@ -0,0 +1,47 @@
+/datum/god_form/starlight/take_charge(var/mob/living/user, var/charge)
+	charge = max(5, charge/100)
+	if(prob(charge))
+		to_chat(user, "<span class='danger'>Your body burns!</span>")
```
```suggestion
		to_chat(user, SPAN_DANGER("Your body burns!"))
```

SierraKomodo commented on code/modules/mob/living/deity/forms/starlight.dm:
```diff
@@ -0,0 +1,47 @@
+				/datum/deity_item/phenomena/divine_right
+				)
+
+/datum/god_form/starlight/take_charge(var/mob/living/user, var/charge)
```
```suggestion
/datum/god_form/starlight/take_charge(mob/living/user, charge)
```

SierraKomodo commented on code/modules/mob/living/deity/forms/starlight.dm:
```diff
@@ -0,0 +1,47 @@
+	if(prob(charge))
+		to_chat(user, "<span class='danger'>Your body burns!</span>")
+	user.adjustFireLoss(charge)
+	return 1
```
```suggestion
	return TRUE
```

SierraKomodo commented on code/modules/mob/living/deity/forms/tower.dm:
```diff
@@ -0,0 +1,49 @@
+	buildables = list(/obj/structure/deity/altar/tower,
+					/obj/structure/deity/pylon,
+					/obj/structure/deity/wizard_recharger
+					)
```
```suggestion
	buildables = list(
		/obj/structure/deity/altar/tower,
		/obj/structure/deity/pylon,
		/obj/structure/deity/wizard_recharger
	)
```

SierraKomodo commented on code/modules/mob/living/deity/forms/tower.dm:
```diff
@@ -0,0 +1,49 @@
+				/datum/deity_item/boon/single_charge/heal/major,
+				/datum/deity_item/boon/single_charge/heal/area,
+				/datum/deity_item/phenomena/rock_form
+				)
```
```suggestion
	items = list(
		/datum/deity_item/general/potential,
		/datum/deity_item/general/regeneration,
		/datum/deity_item/conjuration,
		/datum/deity_item/boon/single_charge/create_air,
		/datum/deity_item/boon/single_charge/acid_spray,
		/datum/deity_item/boon/single_charge/force_wall,
		/datum/deity_item/phenomena/dimensional_locker,
		/datum/deity_item/boon/single_charge/faithful_hound,
		/datum/deity_item/wizard_armaments,
		/datum/deity_item/boon/single_charge/sword,
		/datum/deity_item/boon/single_charge/shield,
		/datum/deity_item/phenomena/portals,
		/datum/deity_item/boon/single_charge/fireball,
		/datum/deity_item/boon/single_charge/force_portal,
		/datum/deity_item/phenomena/banishing_smite,
		/datum/deity_item/transmutation,
		/datum/deity_item/boon/single_charge/slippery_surface,
		/datum/deity_item/boon/single_charge/smoke,
		/datum/deity_item/boon/single_charge/knock,
		/datum/deity_item/boon/single_charge/burning_grip,
		/datum/deity_item/phenomena/warp_body,
		/datum/deity_item/boon/single_charge/jaunt,
		/datum/deity_item/healing_spells,
		/datum/deity_item/boon/single_charge/heal,
		/datum/deity_item/boon/single_charge/heal/major,
		/datum/deity_item/boon/single_charge/heal/area,
		/datum/deity_item/phenomena/rock_form
	)
```

SierraKomodo commented on code/modules/mob/living/deity/forms/tower.dm:
```diff
@@ -0,0 +1,49 @@
+
+/datum/god_form/wizard/take_charge(var/mob/living/user, var/charge)
+	linked_god.adjust_power_min(max(round(charge/100), 1),silent = 1)
+	return 1
```
```suggestion
	return TRUE
```

SierraKomodo commented on code/modules/mob/living/deity/forms/tower.dm:
```diff
@@ -0,0 +1,49 @@
+				/datum/deity_item/phenomena/rock_form
+				)
+
+/datum/god_form/wizard/take_charge(var/mob/living/user, var/charge)
```
```suggestion
/datum/god_form/wizard/take_charge(mob/living/user, charge)
```

SierraKomodo commented on code/modules/mob/living/deity/forms/tower.dm:
```diff
@@ -0,0 +1,49 @@
+				)
+
+/datum/god_form/wizard/take_charge(var/mob/living/user, var/charge)
+	linked_god.adjust_power_min(max(round(charge/100), 1),silent = 1)
```
```suggestion
	linked_god.adjust_power_min(max(round(charge/100), 1), TRUE)
```

SierraKomodo commented on code/modules/mob/living/deity/items/deity_item.dm:
```diff
@@ -0,0 +1,38 @@
+	var/category
+	var/level = 0
+	var/max_level = 0
+	var/list/requirements //Name of item = level of item
```
```suggestion
	/// Name of item = level of item
	var/list/requirements
```

SierraKomodo commented on code/modules/mob/living/deity/items/deity_item.dm:
```diff
@@ -0,0 +1,38 @@
+	var/max_level = 0
+	var/list/requirements //Name of item = level of item
+
+/datum/deity_item/proc/can_buy(var/mob/living/deity/D)
```
```suggestion
/datum/deity_item/proc/can_buy(mob/living/deity/D)
```

SierraKomodo commented on code/modules/mob/living/deity/items/deity_item.dm:
```diff
@@ -0,0 +1,38 @@
+				return FALSE
+	return TRUE
+
+/datum/deity_item/proc/buy(var/mob/living/deity/D)
```
```suggestion
/datum/deity_item/proc/buy(mob/living/deity/D)
```

SierraKomodo commented on code/modules/mob/living/deity/items/deity_item.dm:
```diff
@@ -0,0 +1,38 @@
+	D.adjust_power(-get_cost(D))
+	level++
+
+/datum/deity_item/proc/get_cost(var/mob/living/deity/D)
```
```suggestion
/datum/deity_item/proc/get_cost(mob/living/deity/D)
```

SierraKomodo commented on code/modules/mob/living/deity/items/general.dm:
```diff
@@ -0,0 +1,26 @@
+	desc = "Increase the amount of natural power you regenerate."
+	base_cost = 10
+
+/datum/deity_item/general/potential/buy(var/mob/living/deity/D)
```
```suggestion
/datum/deity_item/general/potential/buy(mob/living/deity/D)
```

SierraKomodo commented on code/modules/mob/living/deity/items/general.dm:
```diff
@@ -0,0 +1,26 @@
+	..()
+	D.adjust_power_min(5)
+
+/datum/deity_item/general/potential/get_cost(var/mob/living/deity/D)
```
```suggestion
/datum/deity_item/general/potential/get_cost(mob/living/deity/D)
```

SierraKomodo commented on code/modules/mob/living/deity/items/general.dm:
```diff
@@ -0,0 +1,26 @@
+	desc = "Decreases the time it takes to charge your power."
+	base_cost = 5
+
+/datum/deity_item/general/regeneration/buy(var/mob/living/deity/D)
```
```suggestion
/datum/deity_item/general/regeneration/buy(mob/living/deity/D)
```

SierraKomodo commented on code/modules/mob/living/deity/items/general.dm:
```diff
@@ -0,0 +1,26 @@
+	..()
+	D.power_per_regen++
+
+/datum/deity_item/general/regeneration/get_cost(var/mob/living/deity/D)
```
```suggestion
/datum/deity_item/general/regeneration/get_cost(mob/living/deity/D)
```

SierraKomodo commented on code/modules/mob/living/deity/items/generic.dm:
```diff
@@ -0,0 +1,25 @@
+/datum/deity_item/boon
+	var/boon_path
+
+/datum/deity_item/boon/buy(var/mob/living/deity/D)
```
```suggestion
/datum/deity_item/boon/buy(mob/living/deity/D)
```

SierraKomodo commented on code/modules/mob/living/deity/items/generic.dm:
```diff
@@ -0,0 +1,25 @@
+	var/phenomena_path
+	max_level = 1
+
+/datum/deity_item/phenomena/buy(var/mob/living/deity/D)
```
```suggestion
/datum/deity_item/phenomena/buy(mob/living/deity/D)
```

SierraKomodo commented on code/modules/mob/living/deity/items/generic.dm:
```diff
@@ -0,0 +1,25 @@
+		D.add_phenomena(phenomena_path)
+		D.update_phenomenas()
+
+/datum/deity_item/boon/single_charge/buy(var/mob/living/deity/D)
```
```suggestion
/datum/deity_item/boon/single_charge/buy(mob/living/deity/D)
```

SierraKomodo commented on code/modules/mob/living/deity/items/narsie/sacrificing.dm:
```diff
@@ -0,0 +1,42 @@
+	base_cost = 50
+	max_level = 1
+
+/datum/deity_item/blood_stone/buy(var/mob/living/deity/user)
```
```suggestion
/datum/deity_item/blood_stone/buy(mob/living/deity/user)
```

SierraKomodo commented on code/modules/mob/living/deity/items/narsie/smithing.dm:
```diff
@@ -0,0 +1,40 @@
+	var/list/recipes = list(/obj/item/melee/cultblade = 50,
+							/obj/item/clothing/head/culthood/alt = 10,
+							/obj/item/clothing/suit/cultrobes/alt = 20
+							)
```
```suggestion
	var/list/recipes = list(
		/obj/item/melee/cultblade = 50,
		/obj/item/clothing/head/culthood/alt = 10,
		/obj/item/clothing/suit/cultrobes/alt = 20
	)
```

SierraKomodo commented on code/modules/mob/living/deity/items/narsie/smithing.dm:
```diff
@@ -0,0 +1,40 @@
+							/obj/item/clothing/suit/cultrobes/alt = 20
+							)
+
+/datum/deity_item/blood_crafting/buy(var/mob/living/deity/user)
```
```suggestion
/datum/deity_item/blood_crafting/buy(mob/living/deity/user)
```

SierraKomodo commented on code/modules/mob/living/deity/items/narsie/smithing.dm:
```diff
@@ -0,0 +1,40 @@
+	requirements = list(DEITY_BLOOD_CRAFT = 1)
+	recipes = list(/obj/item/clothing/suit/cultrobes/magusred = 80,
+					/obj/item/clothing/head/culthood/magus = 50,
+					/obj/structure/constructshell/cult = 70) //also shield?
```
Comment seems unnecessary here

```suggestion
	recipes = list(
		/obj/item/clothing/suit/cultrobes/magusred = 80,
		/obj/item/clothing/head/culthood/magus = 50,
		/obj/structure/constructshell/cult = 70
	)
```

SierraKomodo commented on code/modules/mob/living/deity/items/narsie/smithing.dm:
```diff
@@ -0,0 +1,40 @@
+	base_cost = 100
+	requirements = list(DEITY_BLOOD_CRAFT = 1, DEITY_ARMOR_CRAFT = 1)
+	recipes = list(/obj/item/clothing/suit/space/cult = 100,
+					/obj/item/clothing/head/helmet/space/cult = 70) //Probably more too.
```
```suggestion
	recipes = list(
		/obj/item/clothing/suit/space/cult = 100,
		/obj/item/clothing/head/helmet/space/cult = 70
	)
```

SierraKomodo commented on code/modules/mob/living/deity/items/starlight/artifacts.dm:
```diff
@@ -0,0 +1,32 @@
+	recipes = list(/obj/item/clothing/ring/aura_ring/talisman_of_starborn = 70,
+							/obj/item/clothing/ring/aura_ring/talisman_of_blueforged = 70,
+							/obj/item/clothing/ring/aura_ring/talisman_of_shadowling = 70
+							)
```
```suggestion
	recipes = list(
		/obj/item/clothing/ring/aura_ring/talisman_of_starborn = 70,
		/obj/item/clothing/ring/aura_ring/talisman_of_blueforged = 70,
		/obj/item/clothing/ring/aura_ring/talisman_of_shadowling = 70
	)
```

SierraKomodo commented on code/modules/mob/living/deity/items/tower/conjuration.dm:
```diff
@@ -0,0 +1,107 @@
+	max_level = 3
+	base_cost = 50
+
+/datum/deity_item/conjuration/get_cost(var/mob/living/deity/D)
```
```suggestion
/datum/deity_item/conjuration/get_cost(mob/living/deity/D)
```

SierraKomodo commented on code/modules/mob/living/deity/items/tower/transmutation.dm:
```diff
@@ -0,0 +1,99 @@
+	max_level = 3
+	base_cost = 50
+
+/datum/deity_item/conjuration/get_cost(var/mob/living/deity/D)
```
```suggestion
/datum/deity_item/conjuration/get_cost(mob/living/deity/D)
```

SierraKomodo commented on code/modules/mob/living/deity/menu/deity_nano.dm:
```diff
@@ -0,0 +1,81 @@
+/mob/living/deity
+	var/menu_category
+	var/list/nano_data = list()
+	var/datum/phenomena/selected
```
```suggestion
/mob/living/deity/var/menu_category
/mob/living/deity/var/list/nano_data = list()
/mob/living/deity/var/datum/phenomena/selected
```

SierraKomodo commented on code/modules/mob/living/deity/menu/deity_nano.dm:
```diff
@@ -0,0 +1,81 @@
+	var/list/nano_data = list()
+	var/datum/phenomena/selected
+
+/mob/living/deity/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1, var/uistate = GLOB.self_state)
```
```suggestion
/mob/living/deity/ui_interact(mob/user, ui_key = "main", datum/nanoui/ui, force_open = TRUE, uistate = GLOB.self_state)
```

SierraKomodo commented on code/modules/mob/living/deity/menu/deity_nano.dm:
```diff
@@ -0,0 +1,81 @@
+		ui.open()
+		ui.set_auto_update(TRUE)
+
+/mob/living/deity/proc/set_nano_category(var/num)
```
```suggestion
/mob/living/deity/proc/set_nano_category(num)
```

SierraKomodo commented on code/modules/mob/living/deity/phenomena/communication.dm:
```diff
@@ -0,0 +1,80 @@
+	flags = PHENOMENA_FOLLOWER | PHENOMENA_NONFOLLOWER
+	expected_type = /mob/living
+
+/datum/phenomena/communicate/activate(var/mob/living/L)
```
```suggestion
/datum/phenomena/communicate/activate(mob/living/L)
```

SierraKomodo commented on code/modules/mob/living/deity/phenomena/communication.dm:
```diff
@@ -0,0 +1,80 @@
+		if(!linked.is_follower(L))
+			text_size = 1
+		to_chat(L, "<span class='cult'><font size='[text_size]'>[text_to_send]</font></span>") //Note to self: make this go to ghosties
+		to_chat(linked, "<span class='notice'>You send the message [text_to_send] to \the [L]</span>")
```
```suggestion
		to_chat(linked, SPAN_NOTICE("You send the message [text_to_send] to \the [L]"))
```

SierraKomodo commented on code/modules/mob/living/deity/phenomena/communication.dm:
```diff
@@ -0,0 +1,80 @@
+	expected_type = /atom
+	var/image/arrow
+
+/datum/phenomena/point/activate(var/atom/a)
```
```suggestion
/datum/phenomena/point/activate(atom/a)
```

SierraKomodo commented on code/modules/mob/living/deity/phenomena/communication.dm:
```diff
@@ -0,0 +1,80 @@
+		var/text_size = 4
+		if(!linked.is_follower(L))
+			text_size = 1
+		to_chat(L, "<span class='cult'><font size='[text_size]'>[text_to_send]</font></span>") //Note to self: make this go to ghosties
```
```suggestion
		to_chat(L, SPAN_OCCULT("<font size='[text_size]'>[text_to_send]</font>")) //Note to self: make this go to ghosties
```

SierraKomodo commented on code/modules/mob/living/deity/phenomena/communication.dm:
```diff
@@ -0,0 +1,80 @@
+					addtimer(CALLBACK(src, .proc/remove_image, M), 2 SECONDS)
+
+
+/datum/phenomena/point/proc/remove_image(var/mob/living/L)
```
```suggestion
/datum/phenomena/point/proc/remove_image(mob/living/L)
```

SierraKomodo commented on code/modules/mob/living/deity/phenomena/communication.dm:
```diff
@@ -0,0 +1,80 @@
+	cost = 0
+	flags = PHENOMENA_FOLLOWER
+	expected_type = /mob/living
+	var/static/list/punishment_list = list("Pain (0)" = 0, "Light Wound (5)" = 5, "Brain Damage (10)" = 10, "Heavy Wounds (20)" = 20)
```
```suggestion
	var/static/list/punishment_list = list(
		"Pain (0)" = 0,
		"Light Wound (5)" = 5,
		"Brain Damage (10)" = 10,
		"Heavy Wounds (20)" = 20
	)
```

SierraKomodo commented on code/modules/mob/living/deity/phenomena/communication.dm:
```diff
@@ -0,0 +1,80 @@
+	expected_type = /mob/living
+	var/static/list/punishment_list = list("Pain (0)" = 0, "Light Wound (5)" = 5, "Brain Damage (10)" = 10, "Heavy Wounds (20)" = 20)
+
+/datum/phenomena/punish/activate(var/mob/living/L)
```
```suggestion
/datum/phenomena/punish/activate(mob/living/L)
```

SierraKomodo commented on code/modules/mob/living/deity/phenomena/conjuration.dm:
```diff
@@ -0,0 +1,81 @@
+	cost = 10
+	desc = "Summon a trans-dimensional locker anywhere within your influence. You may transport objects and things, but not people in it."
+
+/datum/phenomena/movable_object/dimensional_locker/activate(var/atom/a, var/mob/living/deity/user)
```
```suggestion
/datum/phenomena/movable_object/dimensional_locker/activate(atom/a, mob/living/deity/user)
```

SierraKomodo commented on code/modules/mob/living/deity/phenomena/conjuration.dm:
```diff
@@ -0,0 +1,81 @@
+	expected_type = /atom
+	var/list/portals = list()
+
+/datum/phenomena/portals/activate(var/atom/a, var/mob/living/deity/user)
```
```suggestion
/datum/phenomena/portals/activate(atom/a, mob/living/deity/user)
```

SierraKomodo commented on code/modules/mob/living/deity/phenomena/conjuration.dm:
```diff
@@ -0,0 +1,81 @@
+		P1.target = get_turf(P2)
+		P2.target = get_turf(P1)
+
+/datum/phenomena/portals/proc/remove_portal(var/portal)
```
```suggestion
/datum/phenomena/portals/proc/remove_portal(portal)
```

SierraKomodo commented on code/modules/mob/living/deity/phenomena/conjuration.dm:
```diff
@@ -0,0 +1,81 @@
+	flags = PHENOMENA_NEAR_STRUCTURE|PHENOMENA_MUNDANE|PHENOMENA_FOLLOWER|PHENOMENA_NONFOLLOWER
+	expected_type = /mob/living
+
+/datum/phenomena/banishing_smite/activate(var/mob/living/L, var/mob/living/deity/user)
```
```suggestion
/datum/phenomena/banishing_smite/activate(mob/living/L, mob/living/deity/user)
```

SierraKomodo commented on code/modules/mob/living/deity/phenomena/conversion.dm:
```diff
@@ -0,0 +1,50 @@
+	flags = PHENOMENA_NONFOLLOWER
+	expected_type = /mob/living
+
+/datum/phenomena/conversion/can_activate(var/atom/target)
```
```suggestion
/datum/phenomena/conversion/can_activate(atom/target)
```

SierraKomodo commented on code/modules/mob/living/deity/phenomena/conversion.dm:
```diff
@@ -0,0 +1,50 @@
+		return 0
+	return 1
+
+/datum/phenomena/conversion/activate(var/mob/living/L)
```
```suggestion
/datum/phenomena/conversion/activate(mob/living/L)
```

SierraKomodo commented on code/modules/mob/living/deity/phenomena/conversion.dm:
```diff
@@ -0,0 +1,50 @@
+	flags = PHENOMENA_NONFOLLOWER
+	expected_type = /mob/living
+
+/datum/phenomena/forced_conversion/can_activate(var/mob/living/L)
```
```suggestion
/datum/phenomena/forced_conversion/can_activate(mob/living/L)
```

SierraKomodo commented on code/modules/mob/living/deity/phenomena/conversion.dm:
```diff
@@ -0,0 +1,50 @@
+
+/datum/phenomena/forced_conversion/can_activate(var/mob/living/L)
+	if(!..())
+		return 0
```
```suggestion
		return FALSE
```

SierraKomodo commented on code/modules/mob/living/deity/phenomena/conversion.dm:
```diff
@@ -0,0 +1,50 @@
+	var/obj/structure/deity/altar/A = locate() in get_turf(L)
+	if(!A || A.linked_god != linked)
+		to_chat(linked,SPAN_WARNING("\The [L] needs to be on \a [linked.get_type_name(/obj/structure/deity/altar)] to be forcefully converted.."))
+		return 0
```
```suggestion
		return FALSE
```

SierraKomodo commented on code/modules/mob/living/deity/phenomena/conversion.dm:
```diff
@@ -0,0 +1,50 @@
+		to_chat(linked,SPAN_WARNING("\The [L] needs to be on \a [linked.get_type_name(/obj/structure/deity/altar)] to be forcefully converted.."))
+		return 0
+
+	return 1
```
```suggestion
	return TRUE
```

SierraKomodo commented on code/modules/mob/living/deity/phenomena/conversion.dm:
```diff
@@ -0,0 +1,50 @@
+
+	return 1
+
+/datum/phenomena/forced_conversion/activate(var/mob/living/L)
```
```suggestion
/datum/phenomena/forced_conversion/activate(mob/living/L)
```

SierraKomodo commented on code/modules/mob/living/deity/phenomena/generic.dm:
```diff
@@ -0,0 +1,36 @@
+	object_to_move = new object_type()
+	GLOB.destroyed_event.register(object_to_move, src, .proc/add_object)
+
+/datum/phenomena/movable_object/activate(var/atom/a, var/mob/living/deity/user)
```
```suggestion
/datum/phenomena/movable_object/activate(atom/a, mob/living/deity/user)
```

SierraKomodo commented on code/modules/mob/living/deity/phenomena/narsie.dm:
```diff
@@ -0,0 +1,36 @@
+	flags = PHENOMENA_FOLLOWER
+	expected_type = /mob/living/carbon/human
+
+/datum/phenomena/exhude_blood/can_activate(var/mob/living/carbon/human/H)
```
```suggestion
/datum/phenomena/exhude_blood/can_activate(mob/living/carbon/human/H)
```

SierraKomodo commented on code/modules/mob/living/deity/phenomena/narsie.dm:
```diff
@@ -0,0 +1,36 @@
+
+/datum/phenomena/exhude_blood/can_activate(var/mob/living/carbon/human/H)
+	if(!..())
+		return 0
```
```suggestion
		return FALSE
```

SierraKomodo commented on code/modules/mob/living/deity/phenomena/narsie.dm:
```diff
@@ -0,0 +1,36 @@
+
+	if(!H.should_have_organ(BP_HEART) || H.vessel.total_volume == H.species.blood_volume)
+		to_chat(linked, SPAN_WARNING("\The [H] doesn't require anymore blood."))
+		return 0
```
```suggestion
		return FALSE
```

SierraKomodo commented on code/modules/mob/living/deity/phenomena/narsie.dm:
```diff
@@ -0,0 +1,36 @@
+	if(!H.should_have_organ(BP_HEART) || H.vessel.total_volume == H.species.blood_volume)
+		to_chat(linked, SPAN_WARNING("\The [H] doesn't require anymore blood."))
+		return 0
+	return 1
```
```suggestion
	return TRUE
```

SierraKomodo commented on code/modules/mob/living/deity/phenomena/narsie.dm:
```diff
@@ -0,0 +1,36 @@
+		return 0
+	return 1
+
+/datum/phenomena/exhude_blood/activate(var/mob/living/carbon/human/H, var/mob/living/deity/user)
```
```suggestion
/datum/phenomena/exhude_blood/activate(mob/living/carbon/human/H, mob/living/deity/user)
```

SierraKomodo commented on code/modules/mob/living/deity/phenomena/narsie.dm:
```diff
@@ -0,0 +1,36 @@
+	expected_type = /mob/living
+	var/static/list/creepy_notes = list("Your knees give out as an unnatural screaming rings your ears.",
+										"You breathe in ash and decay, your lungs gasping for air as your body gives way to the floor.",
+										"An extreme pressure comes over you, as if an unknown force has marked you.")
```
```suggestion
	var/static/list/creepy_notes = list(
		"Your knees give out as an unnatural screaming rings your ears.",
		"You breathe in ash and decay, your lungs gasping for air as your body gives way to the floor.",
		"An extreme pressure comes over you as if an unknown force has marked you."
	)
```

SierraKomodo commented on code/modules/mob/living/deity/phenomena/narsie.dm:
```diff
@@ -0,0 +1,36 @@
+										"You breathe in ash and decay, your lungs gasping for air as your body gives way to the floor.",
+										"An extreme pressure comes over you, as if an unknown force has marked you.")
+
+/datum/phenomena/hellscape/activate(var/mob/living/L)
```
```suggestion
/datum/phenomena/hellscape/activate(mob/living/L)
```

SierraKomodo commented on code/modules/mob/living/deity/phenomena/phenomena.dm:
```diff
@@ -0,0 +1,75 @@
+	var/desc = "This has no desc."
+	var/cost = 0
+	var/mob/living/deity/linked
+	var/flags = 0
```
```suggestion
	var/flags = EMPTY_BITFIELD
```

SierraKomodo commented on code/modules/mob/living/deity/phenomena/phenomena.dm:
```diff
@@ -0,0 +1,75 @@
+	var/refresh_time = 0
+	var/expected_type
+
+/datum/phenomena/New(var/master)
```
```suggestion
/datum/phenomena/New(master)
```

SierraKomodo commented on code/modules/mob/living/deity/phenomena/phenomena.dm:
```diff
@@ -0,0 +1,75 @@
+	linked.remove_phenomena(src)
+	return ..()
+
+/datum/phenomena/proc/Click(var/atom/target)
```
```suggestion
/datum/phenomena/proc/Click(atom/target)
```

SierraKomodo commented on code/modules/mob/living/deity/phenomena/phenomena.dm:
```diff
@@ -0,0 +1,75 @@
+		refresh_time = world.time + cooldown
+		activate(target)
+
+/datum/phenomena/proc/can_activate(var/atom/target)
```
```suggestion
/datum/phenomena/proc/can_activate(atom/target)
```

SierraKomodo commented on code/modules/mob/living/deity/phenomena/phenomena.dm:
```diff
@@ -0,0 +1,75 @@
+
+/datum/phenomena/proc/can_activate(var/atom/target)
+	if(!linked)
+		return 0
```
```suggestion
		return FALSE
```

SierraKomodo commented on code/modules/mob/living/deity/phenomena/phenomena.dm:
```diff
@@ -0,0 +1,75 @@
+		return 0
+	if(refresh_time > world.time)
+		to_chat(linked, SPAN_WARNING("\The [src] is still on cooldown for [round((refresh_time - world.time)/10)] more seconds!"))
+		return 0
```
```suggestion
		return FALSE
```

SierraKomodo commented on code/modules/mob/living/deity/phenomena/phenomena.dm:
```diff
@@ -0,0 +1,75 @@
+
+	if(!linked.form)
+		to_chat(linked, SPAN_WARNING("You must choose your form first!"))
+		return 0
```
```suggestion
		return FALSE
```

SierraKomodo commented on code/modules/mob/living/deity/phenomena/phenomena.dm:
```diff
@@ -0,0 +1,75 @@
+		return 0
+
+	if(expected_type && !istype(target,expected_type))
+		return 0
```
```suggestion
		return FALSE
```

SierraKomodo commented on code/modules/mob/living/deity/phenomena/phenomena.dm:
```diff
@@ -0,0 +1,75 @@
+	if(flags & PHENOMENA_NEAR_STRUCTURE)
+		if(!linked.near_structure(target, 1))
+			to_chat(linked, SPAN_WARNING("\The [target] needs to be near a holy structure for your powers to work!"))
+			return 0
```
```suggestion
			return FALSE
```

SierraKomodo commented on code/modules/mob/living/deity/phenomena/phenomena.dm:
```diff
@@ -0,0 +1,75 @@
+		if(!L.mind || !L.client)
+			if(!(flags & PHENOMENA_MUNDANE))
+				to_chat(linked, SPAN_WARNING("\The [L]'s mind is too mundane for you to influence."))
+				return 0
```
```suggestion
				return FALSE
```

SierraKomodo commented on code/modules/mob/living/deity/phenomena/phenomena.dm:
```diff
@@ -0,0 +1,75 @@
+			if(linked.is_follower(target, silent = 1))
+				if(!(flags & PHENOMENA_FOLLOWER))
+					to_chat(linked, SPAN_WARNING("You can't use [name] on the flock!"))
+					return 0
```
```suggestion
					return FALSE
```

SierraKomodo commented on code/modules/mob/living/deity/phenomena/phenomena.dm:
```diff
@@ -0,0 +1,75 @@
+					return 0
+			else if(!(flags & PHENOMENA_NONFOLLOWER))
+				to_chat(linked, SPAN_WARNING("You can't use [name] on non-believers."))
+				return 0
```
```suggestion
				return FALSE
```

SierraKomodo commented on code/modules/mob/living/deity/phenomena/phenomena.dm:
```diff
@@ -0,0 +1,75 @@
+
+	if(cost > linked.power)
+		to_chat(linked, SPAN_WARNING("You need more power to use [name] (Need [cost] power, have [linked.power])!"))
+		return 0
```
```suggestion
		return FALSE
```

SierraKomodo commented on code/modules/mob/living/deity/phenomena/phenomena.dm:
```diff
@@ -0,0 +1,75 @@
+		to_chat(linked, SPAN_WARNING("You need more power to use [name] (Need [cost] power, have [linked.power])!"))
+		return 0
+
+	return 1
```
```suggestion
	return TRUE
```

SierraKomodo reviewed (changes requested):
Style/formatting

Don't have the energy to go through the remaining 30 files.

SierraKomodo commented on code/modules/mob/living/deity/forms/narsie.dm:
```diff
@@ -0,0 +1,51 @@
+/datum/god_form/narsie/take_charge(mob/living/user, charge)
+	charge = min(100, charge * 0.25)
+	if(prob(charge))
+		to_chat(user, SPAN_WARNING("You feel drained...")
```
Missing a closing `)`

SierraKomodo replied:
Missing a closing `)`

SierraKomodo commented on code/game/gamemodes/godmode/form_items/narsie_structures.dm:
```diff
@@ -0,0 +1,96 @@
+			return
+		user.visible_message(
+			SPAN_NOTICE("\The [user] swirls their hands in \the [src]."),
+			SPAN_NOTIC(text_modifications["Shape"])
```
`SPAN_NOTICE` - Missing the `E`

SierraKomodo reviewed (changes requested):
These look to be the causes of the exploding

SierraKomodo reviewed (dismissed):
It drops when prone now, but runtimes when it drops:

```
[23:00:18] Runtime in norm_aggressive.dm,33: Cannot read null.vars
  proc name: process effect (/datum/grab/normal/aggressive/process_effect)
  src: /datum/grab/normal/aggressive (/datum/grab/normal/aggressive)
  call stack:
  /datum/grab/normal/aggressive (/datum/grab/normal/aggressive): process effect(the grab (upper body) (/obj/item/grab/normal))
  /datum/grab/normal/aggressive (/datum/grab/normal/aggressive): process(the grab (upper body) (/obj/item/grab/normal))
  the grab (upper body) (/obj/item/grab/normal): Process()
  Floella Reade (/mob/living/carbon/human): Life()
  Floella Reade (/mob/living/carbon/human): Life()
  Floella Reade (/mob/living/carbon/human): Life()
  Mobs (/datum/controller/subsystem/mobs): fire(0, null)
  Mobs (/datum/controller/subsystem/mobs): ignite(0)
  Main (/datum/controller/master): RunQueue()
  Main (/datum/controller/master): Loop()
  Main (/datum/controller/master): StartProcessing(0)
```

juxtaposed9930 commented:
> 

what

SierraKomodo reviewed (dismissed):
Merge conflicts alongside whatever's still breaking tests.

MuckerMayhem commented:
The wrong version of the `deity.dm` in `/_defines/` was pushed, looks like it doesn't contain the new define that is seen in the errors.

juxjux9930 committed:
Deity


Sierra Updates

Co-authored-by: Sierra Brown <sierrakomodo@hey.com>
Update code/_onclick/hud/deity.dm

Co-authored-by: Sierra Brown <sierrakomodo@hey.com>
Update code/_onclick/hud/deity.dm

Co-authored-by: Sierra Brown <sierrakomodo@hey.com>
Update code/_onclick/hud/deity.dm

Co-authored-by: Sierra Brown <sierrakomodo@hey.com>
Update code/_onclick/hud/deity.dm

Co-authored-by: Sierra Brown <sierrakomodo@hey.com>
Update code/modules/mob/living/deity/phenomena/phenomena.dm

Co-authored-by: Sierra Brown <sierrakomodo@hey.com>
Update code/modules/mob/living/deity/phenomena/phenomena.dm

Co-authored-by: Sierra Brown <sierrakomodo@hey.com>
Update code/modules/mob/living/deity/phenomena/phenomena.dm

Co-authored-by: Sierra Brown <sierrakomodo@hey.com>
Update code/modules/mob/living/deity/phenomena/phenomena.dm

Co-authored-by: Sierra Brown <sierrakomodo@hey.com>
Update code/modules/mob/living/deity/phenomena/phenomena.dm

Co-authored-by: Sierra Brown <sierrakomodo@hey.com>
Update code/datums/extensions/deity_be_near.dm

Co-authored-by: Sierra Brown <sierrakomodo@hey.com>
Apply first 50 suggestion batch

Co-authored-by: Sierra Brown <sierrakomodo@hey.com>
Apply suggestions from code review

Co-authored-by: Sierra Brown <sierrakomodo@hey.com>
2nd 50 suggestions

Co-authored-by: Sierra Brown <sierrakomodo@hey.com>
More

Co-authored-by: Sierra Brown <sierrakomodo@hey.com>
3rd

Co-authored-by: Sierra Brown <sierrakomodo@hey.com>
Apply suggestions from code review

Co-authored-by: Sierra Brown <sierrakomodo@hey.com>

juxtaposed9930 commented:
oh bruh i think it works now

juxtaposed9930 commented:
is alive

MuckerMayhem commented:
Still a merge conflict

juxtaposed9930 commented:
Rebased again.

SierraKomodo reviewed (changes requested):
See #32084

juxjux9930 committed:
mucker patch

juxtaposed9930 commented:
the muckdate has broken things. until further notice blame me

MuckerMayhem commented:
some more insights from the test Jux and I did:
- runtime occurs when dragging and dropping mobs onto the altar
- spells can be 'spammed' causing runtimes
- there is no notif for the god when a follower does actions to a sacrifice on an altar

MuckerMayhem commented:
Closing as Jux isn't doing this any longer.
